---
lecture_num: 7
scripts:
    - https://cdn.plot.ly/plotly-latest.min.js
---

<section>
    <h2>House-keeping</h2>

    <ol class="spaced" style="margin-top:10%">
        <li>Lab this Friday. Details to be announced.</li>
        <li>Assignment is due <b>Friday this week</b>.</li>
        <li>My office is room 467 (fourth floor) of building 303S (in the CS department)</li>
        <li>Office hours:
            <ul>
                <li>1-3pm, Wednesday</li>
                <li>Almost any other time, but ...</li>
                <li>Email <a href="mailto:tgvaughan@gmail.com">tgvaughan@gmail.com</a> to avoid dissapointment.</li>
            </ul>
        </li>
    </ol>
</section>

<section>
    <h2> Problems with parsimony </h2>

    <ul>
        <li> "Large parsimony problem" is computationally expensive
            <ul>
                <li>Address using heuristic search algorithms.</li>
            </ul>
        </li>

        <li> Gives point estimates: no acknowledgment of uncertainty.</li>

        <li>Questionable biological basis
        <li> Other hidden problems</li>
    </ul>
</section>

<section>
    <section class="center">
        <h2> Modelling neutral sequence evolution</h2>
    </section>

    <section>
        <h3>Why use models?</h3>

        <blockquote style="margin-top:10%">
            We need a model to relate what we observe (data) to what we want to
            know (hypotheses and parameters).
        </blockquote>

        <div style="margin-top:2em"><b>Inference is not possible without a model!</b></div>

        <p style="margin-top:2em">We almost invariably choose
        <b>probabilistic</b> models for molecular evolution, because we don't
        know enough about mutation to construct a deterministic model.</p>
    </section>

    <section>
        <h3>Modelling genetic change</h3>

        <ul class="spaced" style="width:80%;margin-top:5%">
            <li>Given two or more aligned nucleoide or amino acid sequences,
                usually the firs goal is to calculate some measure of sequence
                similarity (or conversely distance).</li>

            <li>The easiest distance to compute is the <b>p-distance</b>: the number of
                differences between two aligned sequences relative to their length.
                <ul>
                    <li>The p-distance is the <b>Hamming distance</b> normalized by the
                        length of the sequence.  Therefore it is the proportion of positions
                        at which the sites differ.</li>
                    <li>The p-distance can also be considered the probability that the two
                        sequences differ at a random site.</li>
                </ul>
            </li>
        </ul>
    </section>

    <section>
        <h3>Modelling genetic change</h3>

        <img data-src="pdist.svg" style="width:80%;margin-top:2em">
    </section>

    <section>
        <h3>p-distance</h3>

        <div style="width:100%; margin-top:10%">
            <img data-src="pdist2.svg" style="vertical-align:middle;width:30%">
            <div style="width: 40%;display:inline-block">
                $p$-distance = 3/7 $\simeq 0.43$.
            </div>
        </div>

        <ul class="spaced" style="width:80%;margin-top:2em">
            <li>Proportion of differences between two sequences.</li>
            <li>Usually <b>understimates</b> the true distance: <b>genetic</b> (or <b>evolutionary</b>) <b>distance</b> $d$.</li>
        </ul>
    </section>

    <section>
        <h3>Multiple, parallel and back-substitutions</h3>

        <img data-src="pdist3.svg" style="width:80%">
    </section>

    <section>
        <h3>Relationship between $p$-distance and genetic distance</h3>
        <div id="pdistOutput" style="width:1000px;height:600px;margin-top:2em">
        <script>

            var trace1 = { x: [], y: [],
                mode: 'lines',
                name: '',
                line: {
                    width: 4,
                    dash: 'solid'
                }
            };

            var trace2 = {x: [], y: [], mode: 'lines', name: '',
                line: { width: 2, dash: 'dash' }
            }
            for (var i=0; i<100; i++) {
                var ut = 3.0*i/100;
                trace1.x[i] = ut;
                trace1.y[i] = 0.75*(1-Math.exp(-4/3*ut));

                trace2.x[i] = ut; trace2.y[i] = ut;
            }

            var data = [trace1, trace2];

            var layout = {
                showlegend: false,
                xaxis: {
                    title: "Genetic distance (d)"
                },
                yaxis: {
                    title: "p-distance (p)",
                    range: [0,1]
                },
                margin: {
                    t:10
                }
            }

            Plotly.newPlot('pdistOutput', data, layout, {displayModeBar: false});
        </script>
        </div>
    </section>


    <section>
        <h3>Continuous-time Markov chains (CTMCs)</h3>

        <ul class="spaced" style="width:80%">
            <li>CTMCs are the continuous-time generalization of Markov chains: state $X(t)$ is
            a function of a continuous time parameter.</li>
            <li>Obey the Chapman-Kolmogorov equation:
                $$p(x_1,t_1|x_0,t_0)=\sum_{x_{1/2}}p(x_1,t_1|x_{1/2},t_{1/2})p(x_{1/2},t_{1/2}|x_0,t_0)$$
                where $p(x_1,t_1)\equiv P(X(t_1)=x_1)$ and $t_0&lt;t_{1/2}&lt;t_1$.
            </li>
            <li>Can be written in the following differential form:
                \begin{align*}
                \frac{d}{dt}p(x,t|x_0,t_0)&amp;=\sum_{x'\neq x}\left[p(x',t|x_0,t_0)Q_{xx'}-p(x,t|x_0,_0)Q_{x'x}\right]\\
                &amp;=\sum_{x'}Q_{x'x}p(x',t|x_0,t_0)
                \end{align*}
                defining $Q_{xx}=-\sum_{x'\neq x}Q_{x'x}$.
            </li>
            <li>This form is known as the Master Equation or Kolmogorov forward equation. Linear, so solution is "simply" $\vec{p}(t)=\exp[-Qt]\vec{p}(0)$.</li>
        </ul>
    </section>

    <section>
        <h3>CTMC example</h3>

        <p>Consider a system with two states (0 and 1) and a transition rate matrix
        $$Q = \left[\begin{array}{cc}
        - &amp; 2 \\
        1 &amp; -
        \end{array}\right]$$</p>

        <p>Gives rise to the following trajectories:</p>

        <div class="textbox">
            <div class="title"><a href="#" style="color:white" id="ctmcRun">Run Demo</a></div>

        <div id="ctmcOutput" style="width:100%;height:200px">
        <script>
            $("#ctmcRun").click(function (e) {
                e.preventDefault();

                console.log("Hello, world!");

                var trace = {
                    x: [0], y: [0],
                    mode: 'lines',
                    line: {shape: 'hv'}};

                var t=0;
                var i=0;
                while (true) {
                    if (trace.y[i]==0)
                        a = 1;
                    else
                        a = 2;
                    
                    t += -Math.log(Math.random())/a;

                    if (t>20)
                        break;

                    trace.x[i+1] = t;
                    trace.y[i+1] = 1 - trace.y[i];

                    i += 1;
                }

                trace.x[i+1] = 20;
                trace.y[i+1] = trace.y[i];

                var layout = {
                    showlegend: false,

                    xaxis: {
                        title: "Time"
                    },
                    yaxis: {
                        title: "State",
                        range: [-0.1,1.1],
                        zeroline: false,
                        tickvals: [0,1]
                    },
                    margin: {
                        t: 20
                    }
                }

                console.log(trace);
                
                Plotly.purge("ctmcOutput");
                Plotly.newPlot("ctmcOutput", [trace], layout, {displayModeBar: false});
            });
        </script>
        </div>
        </div>

        <p>Time $\Delta t$ spent in state before transition:
        $$ P(\Delta t|x) = \lambda e^{-\lambda \Delta t}$$
        where $\lambda = \sum_{x'\neq x}M_{x'x}$.</p>
    </section>

    <section>
        <h3>Jukes-Cantor model of nucelotide substitution</h3>

        <div style="width:100%">
        <img data-src="JC.svg" style="width:30%;vertical-align:middle">
        <div style="display:inline-block;width:50%">
        $$
        Q = \mu\left[\begin{array}{cccc}
        -1 &amp; \frac{1}{3} &amp; \frac{1}{3} &amp; \frac{1}{3}\\
        \frac{1}{3} &amp; -1 &amp; \frac{1}{3} &amp; \frac{1}{3}\\
        \frac{1}{3} &amp; \frac{1}{3} &amp; -1 &amp; \frac{1}{3}\\
        \frac{1}{3} &amp; \frac{1}{3} &amp; \frac{1}{3} &amp;-1
        \end{array}\right]
        $$
        </div>
        </div>

        <div style="margin-top:2em">
            \begin{align*}
            P(X(t)=x|X(0)=x_0) &amp;= \left[\exp(-Qt)\right]_{xx_0}\\
            &amp;=\left\{\begin{array}{ll}
            \frac{1}{4}+\frac{3}{4}e^{-\frac{4}{3}\mu t} &amp; \text{for } x=x_0\\
            \frac{1}{4}-\frac{1}{4}e^{-\frac{4}{3}\mu t} &amp; \text{for } x\neq x_0
            \end{array}\right.
            \end{align*}
        </div>
    </section>

    <section>
        <h3>Differences per site</h3>
        <div id="diffOutput" style="width:1000px;height:600px;margin-top:2em">
        <script>

            var trace1 = { x: [], y: [],
                mode: 'lines',
                name: '',
                line: {
                    width: 4,
                    dash: 'solid'
                }
            };

            var trace2 = {x: [], y: [], mode: 'lines', name: '',
                line: { width: 2, dash: 'dash' }
            }

            var trace3 = {x: [], y: [], mode: 'lines', name: '',
                line: { width: 2, dash: 'dash' }
            }

            for (var i=0; i<100; i++) {
                var ut = 2.0*i/100;
                trace1.x[i] = ut;
                trace1.y[i] = 0.75*(1-Math.exp(-4/3*ut));

                trace2.x[i] = ut; trace2.y[i] = ut;
                trace3.x[i] = ut; trace3.y[i] = 0.75;
            }

            var data = [trace1, trace2, trace3];

            var layout = {
                showlegend: false,
                xaxis: {
                    title: "Expected number of mutations"
                },
                yaxis: {
                    title: "Expected number of differences",
                    range: [0,1]
                },
                margin: {
                    t:10
                }
            }

            Plotly.newPlot('diffOutput', data, layout, {displayModeBar: false});
        </script>
        </div>
    </section>

    <section>
        <h3>Genetic distance under Jukes-Cantor</h3>

        <p>The probabilty that a site is distinct after time $t$ is
        \begin{align*}
        p_{\text{diff}}&amp;=\sum_{x'\neq x} P(X(t)=x'|X(0)=x)\\
        &amp;=\frac{3}{4} - \frac{3}{4}e^{-\frac{4}{3}\mu t}
        \end{align*}
        </p>

        <p style="margin-top:2em">Equating $p_{\text{diff}}$ with the
        $p$-distance and solving for $\mu t$ yields
        $$\hat{d} = \widehat{\mu t} = -\frac{3}{4}\log(1-\frac{4}{3}p).$$
        </p>

        <div style="width:100%">
            <div style="display:inline-block;width:50%;vertical-align:middle">
                <ul>
                    <li>Accounts for multiple, parallel, back-substitutions.</li>
                    <li>Only correct if $p$ is exact: infinite sequences.</li>
                    <li>Diverges if $p&geq;3/4$.</li>
                </ul>
            </div>
            <div style="display:inline-block;width:450px;height:300px;vertical-align:middle" id="distancePlot">
                <script>
            var trace = { x: [], y: [], mode: "lines"};

            for (var i=0; i<100; i++) {
                var p = 0.7*i/100;
                var d = -0.75*Math.log(1-4/3*p);
                trace.x[i] = p;
                trace.y[i] = d;
            }

            var layout = {
                showlegend: false,
                xaxis: {
                    title: "p-distance"
                },
                yaxis: {
                    title: "Genetic distance (Jukes-Cantor)",
                },
                margin: {
                    t:10
                }
            }

            Plotly.newPlot('distancePlot', [trace], layout, {displayModeBar: false});
            </script>
            </div>
        </div>
    </section>


    <section>
        <h3>Other CTMC substitution models: Kimura 80</h3>

        <div style="width:100%">
        <img data-src="K80.svg" style="width:30%;vertical-align:middle">
        <div style="display:inline-block;width:50%">
        $$
        Q = \left[\begin{array}{cccc}
        \cdot &amp; \alpha &amp; \beta &amp; \beta \\
        \alpha &amp; \cdot &amp; \beta &amp; \beta \\
        \beta &amp; \beta &amp; \cdot &amp; \alpha \\
        \beta &amp; \beta &amp; \alpha &amp;\cdot
        \end{array}\right]
        $$
        </div>
        </div>

        <ul style="margin-top:2em">
            <li>Also "Kimura two parameter" model.</li>
            <li>Allows different rates for transitions and transversions.</li>
            <li>Equilibrium base frequencies equal.</li>
        </ul>
    </section>

    <section>
        <h3>Other CTMC substitution models: HKY</h3>

        <div style="width:100%">
        <img data-src="HKY.svg" style="width:30%;vertical-align:middle">
        <div style="display:inline-block;width:50%">
        $$
        Q = \left[\begin{array}{cccc}
        \cdot &amp; \alpha\pi_G &amp; \beta\pi_G &amp; \beta\pi_G \\
        \alpha\pi_A &amp; \cdot &amp; \beta\pi_A &amp; \beta\pi_A \\
        \beta\pi_C &amp; \beta\pi_C &amp; \cdot &amp; \alpha\pi_C \\
        \beta\pi_T &amp; \beta\pi_T &amp; \alpha\pi_T &amp;\cdot
        \end{array}\right]
        $$
        </div>
        </div>

        <ul style="margin-top:2em">
            <li>Due to Hasegawa, Kishino, Yano (1985).</li>
            <li>Allows different transition transversion rates.</li>
            <li>Allows equilibrium base frequencies to differ.</li>
            <li>Most complex model for which analytical solutions are available.</li>
        </li>
    </section>

    <section>
        <h3>Other CTMC substitution models: GTR</h3>

        <div style="width:100%">
        <img data-src="GTR.svg" style="width:30%;vertical-align:middle">
        <div style="display:inline-block;width:50%">
        $$
        Q = \left[\begin{array}{cccc}
        \cdot &amp; \alpha\pi_G &amp; \beta\pi_G &amp; \gamma\pi_G \\
        \alpha\pi_A &amp; \cdot &amp; \delta\pi_A &amp; \epsilon\pi_A \\
        \beta\pi_C &amp; \delta\pi_C &amp; \cdot &amp; \eta\pi_C \\
        \gamma\pi_T &amp; \epsilon\pi_T &amp; \eta\pi_T &amp;\cdot
        \end{array}\right]
        $$
        </div>
        </div>

        <ul style="margin-top:2em">
            <li>All models so-far have been reversible.</li>
            <li>GTR is the most general time-reversible model.</li>
            <li>Reversibility is a mathematical convenience: no biological reason!</li>
            <li>No useful analytical solution available (matrix exponential doesn't count).</li>
        </li>
    </section>


    <section>
        <h3>Genetic distance variation between models</h3>
    </section>

    <section>
        <h3> </h3>
    </section>
</section>

<section>
    <h2>Least-squares Phylogenetic Inference</h2>
</section>

<section>
    <h2>Doing things properly: The Likelihood Function</h2>
</section>

<section>
    <h2>Maximum likelihood inference</h2>

    Sensible methods of parameter estimation from empirical data often
    correspond to values which maximize the likelihood function.
</section>

<section>
    <h2> What is the likelihood of a tree?</h2>
</section>

<section>
    <h2> Felsenstein's Pruning Algorithm</h2>
</section>

<section>
    <h2>Maximum likelihood for Phylogenetics</h2>

    <ul>
        <li>PhyML</li>
        <li>RAxML</li>
        <li>FastTree</li>
    </ul>
</section>

<section>
    <h2>Problems with ML</h2>
</section>

<section>
    <h2>Summary</h2>
</section>

<section>
    <h2>Recommended reading</h3>

    <div class="figure" style="display:inline-block;height:400px;margin-top:2em">
        <img data-src="InferringPhylogenies.jpg">
    </div>

    <div class="figure" style="display:inline-block;height:400px;margin-left:2em;margin-top:2em">
        <img data-src="beast_book_cover.jpeg">
    </div>


</section>
