---
lecture_num: 7
---

<section>
    <h2> Problems with parsimony </h2>

    <img data-src="../lecture6/Parsimony.png" style="width:60%">

    <ul class="spaced" style="margin-top:1em;width:70%">
        <li> "Large parsimony problem" is computationally expensive
            <ul>
                <li>Address using heuristic search algorithms.</li>
            </ul>
        </li>

        <li> Gives point estimates: no acknowledgment of uncertainty.</li>

        <li>Questionable biological basis</li>

        <li> Other hidden problems (such as those <i>ad hoc</i>ery always gives rise to!)</li>
    </ul>
</section>

<section>
    <section class="center">
        <h2> Modelling neutral sequence evolution</h2>
    </section>

    <section>
        <h3>Why use models?</h3>

        <blockquote style="margin-top:10%">
            We need a model to relate what we observe (data) to what we want to
            know (hypotheses and parameters).
        </blockquote>

        <div style="margin-top:2em"><b>Inference is not possible without a model!</b></div>

        <p style="margin-top:2em">We almost invariably choose
        <b>probabilistic</b> models for molecular evolution, because we don't
        know enough about mutation to construct a deterministic model.</p>
    </section>

    <section>
        <h3>Modelling genetic change</h3>

        <ul class="spaced" style="width:80%;margin-top:5%">
            <li>Given two or more aligned nucleoide or amino acid sequences,
                usually the firs goal is to calculate some measure of sequence
                similarity (or conversely distance).</li>

            <li>The easiest distance to compute is the <b>p-distance</b>: the number of
                differences between two aligned sequences relative to their length.
                <ul>
                    <li>The p-distance is the <b>Hamming distance</b> normalized by the
                        length of the sequence.  Therefore it is the proportion of positions
                        at which the sites differ.</li>
                    <li>The p-distance can also be considered the probability that the two
                        sequences differ at a random site.</li>
                </ul>
            </li>
        </ul>
    </section>

    <section>
        <h3>Modelling genetic change</h3>

        <img data-src="pdist.svg" style="width:80%;margin-top:2em">
    </section>

    <section>
        <h3>p-distance</h3>

        <div style="width:100%; margin-top:10%">
            <img data-src="pdist2.svg" style="vertical-align:middle;width:30%">
            <div style="width: 40%;display:inline-block">
                $p$-distance = 3/7 $\simeq 0.43$.
            </div>
        </div>

        <ul class="spaced" style="width:80%;margin-top:2em">
            <li>Proportion of differences between two sequences.</li>
            <li>Usually <b>understimates</b> the true distance: <b>genetic</b> (or <b>evolutionary</b>) <b>distance</b> $d$.</li>
        </ul>
    </section>

    <section>
        <h3>Multiple, parallel and back-substitutions</h3>

        <img data-src="pdist3.svg" style="width:80%">
    </section>

    <section>
        <h3>Relationship between $p$-distance and genetic distance</h3>
        <div id="pdistOutput" style="width:1000px;height:600px;margin-top:2em">
        <script>

            var trace1 = { x: [], y: [],
                mode: 'lines',
                name: '',
                line: {
                    width: 4,
                    dash: 'solid'
                }
            };

            var trace2 = {x: [], y: [], mode: 'lines', name: '',
                line: { width: 2, dash: 'dash' }
            }
            for (var i=0; i<100; i++) {
                var ut = 3.0*i/100;
                trace1.x[i] = ut;
                trace1.y[i] = 0.75*(1-Math.exp(-4/3*ut));

                trace2.x[i] = ut; trace2.y[i] = ut;
            }

            var data = [trace1, trace2];

            var layout = {
                showlegend: false,
                xaxis: {
                    title: "Genetic distance (d)"
                },
                yaxis: {
                    title: "p-distance (p)",
                    range: [0,1]
                },
                margin: {
                    t:10
                }
            }

            Plotly.newPlot('pdistOutput', data, layout, {displayModeBar: false});
        </script>
        </div>
    </section>


    <section>
        <h3>Continuous-time Markov chains (CTMCs)</h3>

        <ul class="spaced" style="width:80%">
            <li>CTMCs are the continuous-time generalization of Markov chains: state $X(t)$ is
            a function of a continuous time parameter.</li>
            <li>Obey the Chapman-Kolmogorov equation:
                $$p(x_1,t_1|x_0,t_0)=\sum_{x_{1/2}}p(x_1,t_1|x_{1/2},t_{1/2})p(x_{1/2},t_{1/2}|x_0,t_0)$$
                where $p(x_1,t_1)\equiv P(X(t_1)=x_1)$ and $t_0&lt;t_{1/2}&lt;t_1$.
            </li>
            <li>Can be written in the following differential form:
                \begin{align*}
                \frac{d}{dt}p(x,t|x_0,t_0)&amp;=\sum_{x'\neq x}\left[p(x',t|x_0,t_0)Q_{xx'}-p(x,t|x_0,_0)Q_{x'x}\right]\\
                &amp;=\sum_{x'}Q_{x'x}p(x',t|x_0,t_0)
                \end{align*}
                defining $Q_{xx}=-\sum_{x'\neq x}Q_{x'x}$.
            </li>
            <li>This form is known as the Master Equation or Kolmogorov forward equation. Linear, so solution is "simply" $\vec{p}(t)=\exp[-Qt]\vec{p}(0)$.</li>
        </ul>
    </section>

    <section>
        <h3>CTMC example</h3>

        <p>Consider a system with two states (0 and 1) and a transition rate matrix
        $$Q = \left[\begin{array}{cc}
        - &amp; 2 \\
        1 &amp; -
        \end{array}\right]$$</p>

        <p>Gives rise to the following trajectories:</p>

        <div class="textbox">
            <div class="title"><a href="#" style="color:white" id="ctmcRun">Run Demo</a></div>

        <div id="ctmcOutput" style="width:100%;height:200px">
        <script>
            $("#ctmcRun").click(function (e) {
                e.preventDefault();

                console.log("Hello, world!");

                var trace = {
                    x: [0], y: [0],
                    mode: 'lines',
                    line: {shape: 'hv'}};

                var t=0;
                var i=0;
                while (true) {
                    if (trace.y[i]==0)
                        a = 1;
                    else
                        a = 2;
                    
                    t += -Math.log(Math.random())/a;

                    if (t>20)
                        break;

                    trace.x[i+1] = t;
                    trace.y[i+1] = 1 - trace.y[i];

                    i += 1;
                }

                trace.x[i+1] = 20;
                trace.y[i+1] = trace.y[i];

                var layout = {
                    showlegend: false,

                    xaxis: {
                        title: "Time"
                    },
                    yaxis: {
                        title: "State",
                        range: [-0.1,1.1],
                        zeroline: false,
                        tickvals: [0,1]
                    },
                    margin: {
                        t: 20
                    }
                }

                console.log(trace);
                
                Plotly.purge("ctmcOutput");
                Plotly.newPlot("ctmcOutput", [trace], layout, {displayModeBar: false});
            });
        </script>
        </div>
        </div>

        <p>Time $\Delta t$ spent in state before transition:
        $$ P(\Delta t|x) = \lambda e^{-\lambda \Delta t}$$
        where $\lambda = \sum_{x'\neq x}Q_{x'x}$.</p>
    </section>

    <section>
        <h3>Jukes-Cantor model of nucelotide substitution</h3>

        <div style="width:100%">
        <img data-src="JC.svg" style="width:30%;vertical-align:middle">
        <div style="display:inline-block;width:50%">
        $$
        Q = \mu\left[\begin{array}{cccc}
        -1 &amp; \frac{1}{3} &amp; \frac{1}{3} &amp; \frac{1}{3}\\
        \frac{1}{3} &amp; -1 &amp; \frac{1}{3} &amp; \frac{1}{3}\\
        \frac{1}{3} &amp; \frac{1}{3} &amp; -1 &amp; \frac{1}{3}\\
        \frac{1}{3} &amp; \frac{1}{3} &amp; \frac{1}{3} &amp;-1
        \end{array}\right]
        $$
        </div>
        </div>

        <div style="margin-top:2em">
            \begin{align*}
            P(X(t)=x|X(0)=x_0) &amp;= \left[\exp(-Qt)\right]_{xx_0}\\
            &amp;=\left\{\begin{array}{ll}
            \frac{1}{4}+\frac{3}{4}e^{-\frac{4}{3}\mu t} &amp; \text{for } x=x_0\\
            \frac{1}{4}-\frac{1}{4}e^{-\frac{4}{3}\mu t} &amp; \text{for } x\neq x_0
            \end{array}\right.
            \end{align*}
        </div>
    </section>

    <section>
        <h3>Differences per site</h3>
        <div id="diffOutput" style="width:1000px;height:600px;margin-top:2em">
        <script>

            var trace1 = { x: [], y: [],
                mode: 'lines',
                name: '',
                line: {
                    width: 4,
                    dash: 'solid'
                }
            };

            var trace2 = {x: [], y: [], mode: 'lines', name: '',
                line: { width: 2, dash: 'dash' }
            }

            var trace3 = {x: [], y: [], mode: 'lines', name: '',
                line: { width: 2, dash: 'dash' }
            }

            for (var i=0; i<100; i++) {
                var ut = 2.0*i/100;
                trace1.x[i] = ut;
                trace1.y[i] = 0.75*(1-Math.exp(-4/3*ut));

                trace2.x[i] = ut; trace2.y[i] = ut;
                trace3.x[i] = ut; trace3.y[i] = 0.75;
            }

            var data = [trace1, trace2, trace3];

            var layout = {
                showlegend: false,
                xaxis: {
                    title: "Expected number of mutations"
                },
                yaxis: {
                    title: "Expected number of differences",
                    range: [0,1]
                },
                margin: {
                    t:10
                }
            }

            Plotly.newPlot('diffOutput', data, layout, {displayModeBar: false});
        </script>
        </div>
    </section>

    <section>
        <h3>Genetic distance under Jukes-Cantor</h3>

        <p>The probabilty that a site is distinct after time $t$ is
        \begin{align*}
        p_{\text{diff}}&amp;=\sum_{x'\neq x} P(X(t)=x'|X(0)=x)\\
        &amp;=\frac{3}{4} - \frac{3}{4}e^{-\frac{4}{3}\mu t}
        \end{align*}
        </p>

        <p style="margin-top:2em">Equating $p_{\text{diff}}$ with the
        $p$-distance and solving for $\mu t$ yields
        $$\hat{d} = \widehat{\mu t} = -\frac{3}{4}\log(1-\frac{4}{3}p).$$
        </p>

        <div style="width:100%">
            <div style="display:inline-block;width:50%;vertical-align:middle">
                <ul>
                    <li>Accounts for multiple, parallel, back-substitutions.</li>
                    <li>Only correct if $p$ is exact: infinite sequences.</li>
                    <li>Diverges if $p&geq;3/4$.</li>
                </ul>
            </div>
            <div style="display:inline-block;width:450px;height:300px;vertical-align:middle" id="distancePlot">
                <script>
            var trace = { x: [], y: [], mode: "lines"};

            for (var i=0; i<100; i++) {
                var p = 0.7*i/100;
                var d = -0.75*Math.log(1-4/3*p);
                trace.x[i] = p;
                trace.y[i] = d;
            }

            var layout = {
                showlegend: false,
                xaxis: {
                    title: "p-distance"
                },
                yaxis: {
                    title: "Genetic distance (Jukes-Cantor)",
                },
                margin: {
                    t:10
                }
            }

            Plotly.newPlot('distancePlot', [trace], layout, {displayModeBar: false});
            </script>
            </div>
        </div>
    </section>


    <section>
        <h3>Other CTMC substitution models: Kimura 80</h3>

        <div style="width:100%">
        <img data-src="K80.svg" style="width:30%;vertical-align:middle">
        <div style="display:inline-block;width:50%">
        $$
        Q = \left[\begin{array}{cccc}
        \cdot &amp; \alpha &amp; \beta &amp; \beta \\
        \alpha &amp; \cdot &amp; \beta &amp; \beta \\
        \beta &amp; \beta &amp; \cdot &amp; \alpha \\
        \beta &amp; \beta &amp; \alpha &amp;\cdot
        \end{array}\right]
        $$
        </div>
        </div>

        <ul style="margin-top:2em">
            <li>Also "Kimura two parameter" model.</li>
            <li>Allows different rates for transitions and transversions.</li>
            <li>Equilibrium base frequencies equal.</li>
        </ul>
    </section>

    <section>
        <h3>Other CTMC substitution models: HKY</h3>

        <div style="width:100%">
        <img data-src="HKY.svg" style="width:30%;vertical-align:middle">
        <div style="display:inline-block;width:50%">
        $$
        Q = \left[\begin{array}{cccc}
        \cdot &amp; \alpha\pi_G &amp; \beta\pi_G &amp; \beta\pi_G \\
        \alpha\pi_A &amp; \cdot &amp; \beta\pi_A &amp; \beta\pi_A \\
        \beta\pi_C &amp; \beta\pi_C &amp; \cdot &amp; \alpha\pi_C \\
        \beta\pi_T &amp; \beta\pi_T &amp; \alpha\pi_T &amp;\cdot
        \end{array}\right]
        $$
        </div>
        </div>

        <ul style="margin-top:2em">
            <li>Due to Hasegawa, Kishino, Yano (1985).</li>
            <li>Allows different transition transversion rates.</li>
            <li>Allows equilibrium base frequencies to differ.</li>
            <li>Most complex model for which analytical solutions are available.</li>
        </li>
    </section>

    <section>
        <h3>Other CTMC substitution models: GTR</h3>

        <div style="width:100%">
        <img data-src="GTR.svg" style="width:30%;vertical-align:middle">
        <div style="display:inline-block;width:50%">
        $$
        Q = \left[\begin{array}{cccc}
        \cdot &amp; \alpha\pi_G &amp; \beta\pi_G &amp; \gamma\pi_G \\
        \alpha\pi_A &amp; \cdot &amp; \delta\pi_A &amp; \epsilon\pi_A \\
        \beta\pi_C &amp; \delta\pi_C &amp; \cdot &amp; \eta\pi_C \\
        \gamma\pi_T &amp; \epsilon\pi_T &amp; \eta\pi_T &amp;\cdot
        \end{array}\right]
        $$
        </div>
        </div>

        <ul style="margin-top:2em">
            <li>All models so-far have been reversible.</li>
            <li>GTR is the most general time-reversible model.</li>
            <li>Reversibility is a mathematical convenience: no biological reason!</li>
            <li>No useful analytical solution available (matrix exponential doesn't count).</li>
        </li>
    </section>

    <section>
        <h3>Variable rates among sites</h3>

        <div style="width:100%">
            <div style="width:50%;display:inline-block;vertical-align:middle">
                <p>Gamma distribution rate heterogeneity allows for variation
                in evolutionary rate among sites according to
                $$f(r) = \frac{\alpha^{\alpha}}{\Gamma(\alpha)}r^{\alpha-1}e^{-\alpha r}$$</p>
            </div>
            <div style="width:400px;height:300px;display:inline-block;vertical-align:middle" id="gammaDistribs">
            <script>

            function getTrace(alpha) {
                var trace = { x: [], y: [], mode: "lines", name: "alpha=" + alpha };

                for (var i=0; i<100; i++) {
                    var r = 3*i/100;
                    var f = Math.pow(alpha,alpha)*Math.pow(r, alpha-1)*Math.exp(-alpha*r)/Gamma(alpha);
                    trace.x[i] = r;
                    trace.y[i] = f;
                }

                return trace;
            }

            var layout = {
                showlegend: true,
                xaxis: {
                    title: "Rate"
                },
                yaxis: {
                    title: "Site fraction",
                },
                margin: {
                    t:10,
                }
            }

            Plotly.newPlot('gammaDistribs',
                    [getTrace(20),
                    getTrace(5),
                    getTrace(1),
                    getTrace(0.5),
                    getTrace(0.1)],
                    layout, {displayModeBar: false, staticPlot: true});
            </script>
        </div>

        <div style="width:100%">
            <div style="width:50%;display:inline-block;vertical-align:middle">
                <p>Analytical solution exists for JC distance with site rate heterogeneity:
                $$\hat{d}=-\frac{3}{4}\alpha\left[1-\left(1-\frac{4}{3}p\right)^{-1/\alpha}\right]$$
            </div>
            <div style="width:400px;height:300px; display:inline-block;vertical-align:middle" id="variableSiteD">
            <script>
                    var trace1 = { x: [], y: [], mode: "lines", name: "alpha=1"};
                    var traceInf = { x: [], y: [], mode: "lines", name: "alpha=Inf"};

                    for (var i=0; i<100; i++) {
                        var p = 0.5*i/100;
                        var d1 = 0.75*(Math.pow((1-4/3*p), -1)-1);
                        var dInf = -0.75*Math.log(1-4/3*p);
                        trace1.x[i] = p;
                        traceInf.x[i] = p;
                        trace1.y[i] = d1;
                        traceInf.y[i] = dInf;
                    }

            var layout = {
                showlegend: true,
                xaxis: {
                    title: "p-distance"
                },
                yaxis: {
                    title: "Genetic distance",
                },
                margin: {
                    t:10,
                }
            }

            Plotly.newPlot('variableSiteD',
                    [trace1, traceInf],
                    layout, {displayModeBar: false, staticPlot: false});

            </script>
            </div>
        </div>
    </section>


    <section>
        <h3>Genetic distance variation between models</h3>

        <ul class="spaced" style="margin-top:2em">
            <li>HIV-1B vs HIV-O/SIVcpz/HIV-1C: <i>env</i> gene:
                <table>
                    <tr> <td></td> <th>$p$-distance</th> <th>JC69</th> <th>K80</th> <th>Tajima-Nei</th></tr>
                    <tr> <th>HIV-O</th> <td>0.391</td> <td>0.552</td> <td>0.560</td> <td>0.572</td></tr>
                    <tr> <th>SIVcpz</th> <td>0.266</td> <td>0.337</td> <td>0.340</td> <td>0.427</td></tr>
                    <tr> <th>HIV-1C</th> <td>0.163</td> <td>0.184</td> <td>0.187</td> <td>0.189</td></tr>
                </table>
            </li>

            <li>HIV-1B vs HIV-O/HIV-1C: <i>pol</i> gene:
                <table>
                    <tr> <td></td> <th>$p$-distance</th> <th>JC69</th> <th>K80</th> <th>Tajima-Nei</th></tr>
                    <tr> <th>HIV-O</th> <td>0.257</td> <td>0.315</td> <td>0.318</td> <td>0.324</td></tr>
                    <tr> <th>HIV-1C</th> <td>0.103</td> <td>0.111</td> <td>0.113</td> <td>0.114</td></tr>
                </table>
            </li>
    </section>
</section>

<section>
    <h2>Least-squares Phylogenetic Inference</h2>

    <div>
        <div style="display:inline-block;width:45%;vertical-align:middle">
            <ul>
                <li>Can use genetic distance as the basis for least-squares-based phylogenetic inference.</li>
                <li>Define cost of tree as the (weighted) sum of the squares of the
                difference between the genetic distance on the tree and the
                distance estimated from the sequence alignment:
                $$C(T) = \sum_i\sum_j w_{ij}(d_{ij}(T)-\hat{d}_{ij})^2$$ </li>
                <li>Search for tree $T$ with lowest $C(T)$.</li>
            </ul>
        </div>
        <div style="display:inline-block;width:45%;vertical-align:middle">
            <img data-src="lsq.svg" style="width:100%">
        </div>
    </div>

    <p style="margin-top:0">Comments:
    <ul>
        <li>Ignores correlations due to shared ancestry.</li>
        <li>Consistent estimator: converges to truth in the limit of infinitely long alignment.</li>
    </ul></p>
</section>

<section>
    <h2>The Likelihood Function</h2>

    <blockquote style="margin-top:5%">
        The likelihood for a parameter $\theta$ under model $M$ and given data $D$ is defined as
        $L(\theta|D,M)\equiv P(D|\theta,M)$.
    </blockquote>

    <div style="margin-top:1em">
        <b>The likelihood of $\theta$ is NOT a probability distribution over $\theta$!</b>
    </div>

    <p style="margin-top:2em">Example:</p>
    <div style="margin-top:0">
        <div style="display:inline-block;width:50%;vertical-align:middle;margin-top:0">
            <ul>
                <li>Coin tossed 5 times (identical, independent): $D=(H,T,T,H,T)$.</li>
                <li>Probability of sequence:
                    \begin{align*}
                    P(D|f)&amp;=f\times (1-f)\times(1-f)\times f\times(1-f)\\
                    &amp;=f^2(1-f)^3\\
                    &amp;=L(f|D)
                    \end{align*}
                </li>
            </ul>
        </div>
        <div style="display:inline-block;width:45%;vertical-align:middle">
            <div id="likelihoodOut" style="width:450px;height:300px">
            <script>
            var trace = { x: [], y: [], mode: "lines", name: ""};

            for (var i=0; i<100; i++) {
                var f = i/100;
                var L = Math.pow(f,2)*Math.pow(1-f,3);
                trace.x[i] = f;
                trace.y[i] = L;
            }

            var layout = {
                showlegend: false,
                xaxis: {
                    title: "f"
                },
                yaxis: {
                    title: "L(f|D)",
                },
                margin: {
                    t:10,
                }
            }

            Plotly.newPlot('likelihoodOut', [trace], layout, {displayModeBar: false});
            </script>
            </div>
        </div>
    </div>
</section>

<section>
    <h2>Maximum likelihood inference</h2>

    <blockquote>
    Sensible methods of parameter estimation from empirical data often
    correspond to values which maximize the likelihood function.
    </blockquote>

    
    <div id="maxLikelihoodOut" style="width:800px;height:250px;margin-left:100px">
            <script>
            var trace = { x: [], y: [], mode: "lines", name: ""};

            for (var i=0; i<100; i++) {
                var f = i/100;
                var L = Math.pow(f,2)*Math.pow(1-f,3);
                trace.x[i] = f;
                trace.y[i] = L;
            }

            var layout = {
                showlegend: false,
                xaxis: {
                    title: "f"
                },
                yaxis: {
                    title: "L(f|D)",
                },
                margin: {
                    t:10,
                    b:20
                }
            }

            layout.shapes = [ {
                type: 'line',
                x0: 0.4,
                y0: 0,
                x1: 0.4,
                y1: 0.04,
                line: {
                    dash: 'dash',
                    color: 'orange',
                    width: 2
                }
            } ];

            Plotly.newPlot('maxLikelihoodOut', [trace], layout, {displayModeBar: false});
            </script>
    </div>

    <p>Maximum likelihood estimate of coin fairness parameter $f$ for $n$ heads:
    $$\log L(f|D) = n\log f + (N-n)\log(1-f)$$
    $$\left.\frac{\partial}{\partial f}\log L(f|D)\right|_{f=f_{ML}} = \frac{n}{f_{ML}} - \frac{N-n}{1-f_{ML}} = 0$$
    $$\implies f_{ML} = \frac{n}{N}$$
    </p>
</section>

<section>
    <h2> What is the likelihood of a tree?</h2>
    <img data-src="treeLikelihood.svg" style="width:40%">

    <ul>
        <li>$$L(T|A)\equiv P(A|T) = \sum_x\sum_w P(x)P(A|x)P(w|x)P(G|w)P(G|w)$$</li>
        <li>For $n$ taxa there are $4^{n-1}$ possible internal node states!</li>
        <li>Solve using Dynamic Programming!</li>
    </li>
</section>

<section>
    <h2> Felsenstein's Pruning Algorithm</h2>
    <div>
        <div style="width:50%;display:inline-block;vertical-align:middle">
            <ul>
                <li>Introduced by Joseph Felsenstein, 1973.</li>
                <li>Recursion based on the conditional likelihood of a subtree below node $k$ having state $s$: $L_k(s)$.</li>
                <li>For leaves, $L_k(s)=\delta_{s,l}$ where $l$ is the leaf state.</li>
            </ul>
        </div>
        <div style="width:45%;display:inline-block;vertical-align:middle">
            <img data-src="treeLikelihood.svg" style="width:60%">
        </div>

        <p> For internal nodes,
        $$L_k(s) = \left(\sum_xP(x|s)L_{c_l}(x)\right)\left(\sum_yP(y|s)L_{c_r}(y)\right)$$</p>

        <blockquote>
            Time complexity for $m$ sites: $m(n-1)4^2$.  This is the workhorse of computational phylogenetics.
        </blockquote>
    </div>
</section>

<section>
    <h2>Maximum likelihood for Phylogenetics</h2>

    <ul class="spaced" style="margin-top:10%">
        <li>PhyML (Guindon and Gascuel, 2003)
            <ul>
                <li>Hill-climbing algorithm which attempts to find ML trees.</li>
            </ul>
        </li>
        <li>RAxML (e.g. Stamatakis, 2005)
            <ul>
                <li>Another hill-climbing algorithm, optimized for large trees.</li>
            </ul>
        </li>
        <li>FastTree (Price, Dehal, Arkin, 2009)
            <ul>
                <li>Uses efficient Neighbour-Joining algorithm to find starting tree.</li>
                <li>Updates tree using heuristics based on tree length.</li>
                <li>Inference from alignments of $&gt;10^5$ sequences feasible.</li>
            </ul>
        </li>
    </ul>
</section>

<section>
    <h2>Problems with ML</h2>

    <ul class="spaced" style="margin-top:10%">
        <li>Only ever gives point estimates: no indication of uncertainty.
            <ul>
                <li>Addressed using more <i>ad hoc</i>ery: bootstrap algorithms.</li>
            </ul>
        </li>

        <li>Lacks rigorous theoretical basis: $L(\theta|D)$ is not a probability!</li>

        <li>No clear way of incorporating additional (prior?) information into the analysis.</li>
    </ul>

    <blockquote style="margin-top:2em">
        Need a better approach!
    </blockquote>

</section>

<section>
    <h2>Summary</h2>
    
    <ul class="spaced" style="width:80%">
        <li>The genetic distance between sequence pairs can be estimated using
            a CTMC model for neutral molecular evolution.</li>
        <li>Addresses some of the problems with the parsimony approach.</li>
        <li>Have focused on 4 state DNA models, but it is possible to consider
            models for proteins (20 amino acid states) and models that take
            into account <b>codons</b> and the <b>genetic code</b> (64 states).</li>
        <li>Substitution model approaches to estimating genetic distance do not
            deal with insertions and deletions of genetic material: they presuppose
            a sequence alignment.</li>
        <li>Can be used within the context of a least-squares framework to find
            best-fitting phylogenies.</li>
        <li>Better used as the basis for a tree likelihood.</li>
        <li>Likelihood can be used on its own for phylogenetic inference.</li>
    </ul>
</section>

<section>
    <h2>Recommended reading</h3>

    <div class="figure" style="display:inline-block;height:400px;margin-top:2em">
        <img data-src="InferringPhylogenies.jpg">
    </div>

    <div class="figure" style="display:inline-block;height:400px;margin-left:2em;margin-top:2em">
        <img data-src="beast_book_cover.jpeg">
    </div>


</section>
