---
lecture_num: 12
---
<section>
  <h2>The murky boundary between population genetics and phylogenetics</h2>
  <p>There has been increased interest in analyses of closely related species, where the effect of population genetic processes, such as the coalescent can't be ignored.</p>
  <ul>
    <li>Different gene trees can have different topologies due to incomplete lineage sorting</li>
    <li>Divergence times of species can be overestimated due to ancestral polymorphism</li>
    <li>Sometimes the exact species identities of individuals are not known</li>
    <li>Sometimes researchers identify species based on a split in a single gene tree.</li>
  </ul>
  <p>Enter the multi-species coalescent.</p>
</section>

<section>
  <div style="display:flex; justify-content:space-between">
    <div style="width:30%">
      <img src="LineageDrift1.png" style="width:100%">
      <p>All these patterns are the consequence of "lineage drift" within a single population</p>
    </div>
    <div style="width:30%">
      <img src="LineageDrift2.png" style="width:100%">
    </div>
    <div style="width:30%">
      <img src="LineageDrift3.png" style="width:100%">
    </div>
  </div>
</section>

<section>
  <h2>Gene trees and species trees</h2>
  <img src="GeneAndSpeciesTrees.png" style="width:100%">
</section>

<section>
  <h2>Why Ancestral Polymorphisms?</h2>
  <img src="WhyAncestral.png" style="width:80%">
</section>

<section>
  <h2>Probabilities of Monophyly, Paraphyly and Polyphyly</h2>
  <img src="Phyly.png" style="width:100%">  
</section>

<section>
  <h2>Overestimating divergence times</h2>
  <div style="display:flex; justify-content:space-between">
    <div style="width:35%">
      <ul>
        <li>Typically we use gene phylogenies to estimate species phylogenies</li>
        <li>But divergence time for genes will be longer than that of species.</li>
      </ul>
    </div>
    <div style="width:65%">
      <img src="DivergenceTimes1.png" style="width:100%">
    </div>
  </div>
</section>

<section>
  <h2>Overestimating divergence times</h2>
  <img src="DivergenceTimes2.png" style="width:100%">
  <p style="font-size:0.6em">From Edwards and Beerli, 2000, Evolution 54: 1839-1854</p>
</section>

<section>
  <h2>The multispecies coalescent</h2>
  <img src="geneTreeSpeciesTree.png" style="width:100%">
</section>

<section>
  <h2>Population sizes on a species tree</h2>
  <p>Define \(N_i\) to be the effective population size at the present for the species \(i \in \{1,2,\dots,n\}\), 
  and \(A_i\) the ancestral effective population of species \(i\) at the time of the species origin.</p>
  <p>Then for all ancestral species \(j \in \{ n+1, n+2, \dots, 2n\}\) (represented by internal branches in the species tree) 
  define \(N_j = A_{\text{left}(j)} + A_{\text{right}(j)}\), where \(\text{left}(i)\) is the left descendent 
  of species \(i\) and \(\text{right}(i)\) is the right descendent species.</p>
  \[A_k \sim \text{Exp}(\Theta), \quad 1 \leq k \leq 2n\]
  \[N_i \sim \text{Gamma}(2, \Theta), \quad 1 \leq i \leq n\]
  \[N_j = A_{\text{left}(j)} + A_{\text{right}(j)}, \quad n < j < 2n\]
</section>

<section>  
  <h2>Population sizes prior</h2>
  <img src="populationPrior.png" style="width:100%; display:block; margin:auto">
</section>

<section>
  <h2>Species divergence times prior</h2>
  <p>For a species tree of \(n\) species, define \(T_i\) to be the time at which the species tree goes from having \(i\) to \(i-1\) species, back in time. Additionally define \(\tau_i = T_i - T_{i+1}, i \in \{2,\dots,n-1\}\) and \(\tau_n = T_n\).</p>
  <p>The Yule speciation prior supposes a uniform rate species birth (\(\lambda\)) on all lineages, implying a prior of:</p>
  \[\tau_i \sim \text{Exp}(1/i\lambda)\]
  <p>More complex species tree priors that admit species extinction (Birth-death prior; Gernhard, 2008) and incomplete sampling (Birth-death-sampling; Stadler, 2009) are also possible.</p>
  <p>All of these species tree priors imply a uniform prior on labelled histories.</p>  
</section>

<section>
  <h2>Coalescent prior for gene trees</h2>
  <div style="display:flex; justify-content:space-between">
    <div style="width:65%">
      <p>Consider a single species in the species tree, spanned by \(k = u-v\) coalescent intervals (and a final interval without a coalescent event). \(t_k\) is the time during which there are \(k\) lineages. Define \(N(s)\) as the population size of this species at time \(s\). Define \(s_i = \sum_{k=u}^it_k\). The prior density for each interval ending in a coalescent is:</p>
      \[f(t_k) = {\frac{1}{N(s_k)}}{\exp\left(-{\int \limits_{s_{k-1}}^{s_k} \frac{\binom{n}{2}}{N(x)} dx }\right)}\]
    </div>
    <div style="width:35%">
      <img src="onePopulationCoalescent.png" style="width:100%">  
    </div>
  </div>
</section>

<section>
  <h2>Coalescent prior for gene trees</h2>
  <p>The coalescent prior density for the final interval that does not end in a coalescent event is:</p>
  \[f(t_v) = {\exp\left(-{\int \limits_{s_{v-1}}^{s_v} \frac{\binom{n}{2}}{N(x)} dx }\right)}\]
  <p>Define \(f_g(g | S)\) to be the total coalescent density for gene tree \(g\),  obtained from the product of all the intervals over all species in the species tree \((S)\).</p>  
</section>

<section>
  <h2>Posterior probability of multi-species coalescent</h2>
  <p>If we additionally define \(\color{orange}{\Pr(D_g | g)}\) as the phylogenetic <span style="color:orange">likelihood</span> of the sequence data for gene tree \(g\), and \(\color{red}{P(S | \lambda, \Theta)}\) as the <span style="color:red">prior</span> on the species tree times and population sizes, then the <span style="color:#007f7f">posterior distribution</span> over gene trees, species tree and other parameters is:</p>
$${\color{#007f7f}{P(\mathbf{G},S,\lambda,\Theta | \mathbf{D})}} = \frac{1}{\color{green}Z}  \color{orange}{\left[ \prod_{g \in \mathbf{G}} \Pr(D_g|g) \right]} \color{red}{\left[\prod_{g\in\mathbf{G}} P(g|S)\right] P(S | \lambda, \Theta) P(\lambda) P(\Theta)}$$ 
  <p>where \(\mathbf{G}\) is all the gene trees, \(\mathbf{D}\) is all the gene alignments, \(\color{red}{P(\lambda)}\) is the prior for the speciation rate, \(\color{red}{P(\Theta)}\) is the prior on the ancestral population size parameters, and \(\color{green}Z = P(\mathbf{D})\) is the unknown normalising constant.</p>
</section>

<section>
  <h2>Four gene trees inside a 3-species tree</h2>
  <img src="b1_speciesTree.png" style="width:90%; display:block; margin:auto">
</section>  

<section>
  <h2>A rapid radiation of 7 species</h2>
  <div style="display:flex; justify-content:space-between">
    <div style="width:50%">
      <p>Species tree</p>
      <img src="run1_sptree.png" style="width:100%">
    </div>
    <div style="width:50%">  
      <p>Species tree and one gene tree</p>
      <img src="run1_sptreeand1loci.png" style="width:100%">
    </div>
  </div>
</section>

<section>
  <h2>Gene concatenation - a terrible idea</h2>
  <div style="display:flex; justify-content:space-between">
    <div style="width:26%">
      <p style="font-size:0.6em">Species tree</p>
      <img src="sp2.png" style="width:100%">
    </div>
    <div style="width:74%">
      <p style="font-size:0.6em">"Gene" tree from concatenated genes</p>
      <img src="concatTree.png" style="width:100%">
    </div>
  </div>
</section>

<section>  
  <h2>Species tree of Pocket Gophers</h2>
  <p>Data from (Belfiore, 2008)</p>
  <p>27 individuals, 7 loci (12 from T. bottae, 23 from others, 1 from outgroup)</p>
  <img src="go3.png" style="width:70%; display:block; margin:auto; margin-top:20px">
</section>

<section>
  <h2>Full Bayesian inference under the multi-species coalescent</h2>
  <p>Data from (Belfiore, 2008), software implemented in BEAST by Heled and Drummond (2010)</p>
  <img src="go4.png" style="width:65%; display:block; margin:auto">  
</section>

<section>
  <h2>Open questions</h2>
  <ul>
    <li>Are there better priors for the population sizes?</li>
    <li>What are efficient proposal distributions for MCMC on the multispecies coalescent?</li>
    <li>What about uncertain species identification?
      <ul>
        <li>How do we characterize the prior distribution on species associations? (geography, morphology...)</li>
      </ul>
    </li>
    <li>What about uncertain numbers of species?!  
      <ul>
        <li>How do we characterize the hypothesis space over a species trees with a <i>random</i> number of species, and gene tree tips with an uncertain species identity?</li>
        <li>Reversible-jump, yes, but what proprosal distributions? ;-) What about Bayesian stochastic variable selection?</li>
      </ul>
    </li>
  </ul>
</section>
