---
lecture_num: 14
---

<section>
    <h2>Recombination</h2>

    <ul class="spaced" style="width:80%">
        <li>Occurs in eukaryotes, bacteria and viruses.</li>
        <li>Eukaryotic recombination occurs during meiosis via chromosomal crossover.</li>
        <li>Bacterial recombination occurs via
            <ul>
                <li>phage-mediated transduction,</li>
                <li>natural transformation, and</li>
                <li>conjugation.</li>
            </ul>
        </li>
        <li>Viral recombination occurs when multiple strains infect a single
            cell. (Example: reassortment in influenza.)</li>
    </ul>

    <blockquote>
        Extremely important for phylogenetics: affects the validity of the "tree" assumption!
    </blockquote>
</section>

<section>
    <h2>Wright-Fisher model with recombination</h2>

    <ul>
        <li>Consider an WF model with male and female diploid individuals.</li>
        <li>Focus on a small segment of the genome.</li>
    </ul>

    <img data-src="WFrec.svg" style="width:100%">

    <ul>
        <li>Each child selects 1 male and 1 female parent uniformly at random.</li>
        <li>With probability $r$ (which depends on the segment size) the two
            chromosome pairs of a parent are recombined before being passed
            on.</li>
    </ul>
</section>

<section>
    <h2>Wright-Fisher model with recombination</h2>

    <ul>
        <li>Since the specific pairing of chromosomes only matters over a
            single generation, in the long term the haploid approximation is
            very good:</li>
    </ul>

    <img data-src="WFrec_chromosome.svg" style="width:100%;margin-top:2em">

    <ul style="margin-top:1em">
        <li>Each child in $i+1$ selects a parent uniformly at random from
            generation $i$.</li>
        <li>With probability $r$ an additional parent is selected.</li>
        <li>In this case, a breakpoint chosen uniformly on the chromosome, and
            everything to either the left or right replaced by the homologous
            section of the second parent's chromosome.</li>
        <li>(Equivalent to selecting 2 parents for each child and letting the
            resulting sequences recombine with some probability.)
    </ul>
</section>

<section>
    <h2>Coalescent with recombination</h2>

    <div style="width:55%;display:inline-block;vertical-align:middle">
        <p>For fixed recombination rate $\rho=r/g$ in the limit $r\ll 1$, $g\ll 1$
        and $N\gg 1$, the genealogical process is the coalescent with
        recombination (<a class="cite" href="https://doi.org/10.1016/0040-5809(83)90013-8">Hudson, 1983</a>):</p>

        <ul>
            <li>Coalescence rate: $T_c(k)=\binom{k}{2}\frac{1}{Ng}$</li>
            <li>Recombination rate: $T_r(k)=\rho k$.</li>
            <li>Recombination break points chosen uniformly along sequence: everything
            to the left descends from one parent, everything else to the other.</li>
            <li>Each site possess a <b>local tree</b>.</li>
            <li>Local trees may find MRCAs (grey sites) before <b>grand (G)MRCA</b> of the process.</li>
        </ul>
    </div>
    <img data-src="CWR.svg" style="width:40%;display:inline-block;vertical-align:middle">

</section>

<section>
    <h2>CwR as a birth-death process</h2>

    <div style="width:70%;display:inline-block;vertical-align:middle">
    <p>The number of surviving ancestral lineages $k$ evolves under the
    birth-death process corresponding to the following pair of reactions:</p>
    \begin{align*}
    2L &amp;\overset{\chi}{\longrightarrow} L\\
    L &amp;\overset{\rho}{\longrightarrow} 2L
    \end{align*}
    <p>where $\chi=1/Ng$. The deterministic approximation for the evolution of $\langle k\rangle=\langle k\rangle$ is
    \begin{align*}
    \partial_t \langle k\rangle &amp;\simeq -\frac{1}{2N}\langle k\rangle(\langle k\rangle-1) + \rho\langle k\rangle\\
    &amp;\propto \langle k\rangle(1-\langle k\rangle/k_c)
    \end{align*}
    where $k_c=1+2N\rho$.</p>

    <p>$\langle k\rangle$ stabilizes at $k_c$, but noise <i>eventually</i> drives $k$ to 1 (GMRCA).</p>
    </div>

    <div id="potential" style="width:250px;display:inline-block;vertical-align:middle">
    <script>
var N = 1, rho=30;

var kvec=[];
var U = [];
for (var k=0; k<=100; k++) {
    kvec.push(k);
    U.push(k*k*k/(6*N) - 0.5*(rho+1/(2*N))*k*k);
} 

var trace = {
    x: kvec,
    y: U,
    mode: "lines",
    name: "U",
    line: {
        color: "grey",
        width: 2,
    }
};

var layout =  {
    xaxis: { title: "k" },
    yaxis: { title: "U(k)" },
    title: "Effective potential",
    margin: {t:30, l:30, r:30, b:30}
};


Plotly.newPlot("potential", [trace], layout, {displayModeBar: false});

    </script>
    </div>
</section>

<section>
    <h2>CwR example ARG</h2>

    <img data-src="simulated_arg.png" style="width:100%">

    (Simulated using <a href="CoalescentTreeWithRecombination.xml">this MASTER script</a>.)
</section>

<section>
    <h2>Coalescent with gene conversion</h2>
    <div style="width:55%;display:inline-block;vertical-align:middle">
        <p>A gene conversion event refers to the replacement of a single
        contiguous stretch of sequence with a homologous stretch from a
        different parent. Incorporated into a modified CwR process by
        <a class="cite" href="http://www.genetics.org/content/155/1/451.short"> Wiuf&nbsp;and&nbsp;Hein (2000)</a>.</p>
        <ul>
            <li>Allows similar patterns of site ancestry but with fewer events.</li>
            <li>Model applicable to prokaryotic recombination.</li>
            <li>Coalescence and recombination rates as for CwR.</li>
            <li>Conversions initiate at randomly chosen starting sites and
            extend for $d$ sites where
            $P(d)=(1-\delta^{-1})^{d-1}\delta^{-1}$ (geometric distribution).</li>
        </ul>

    </div>
    <img data-src="CWGC.svg" style="width:40%;display:inline-block;vertical-align:middle">

</section>

<section>
    <h2>Sequentially Markov Coalescent (SMC)</h2>
    <ul>
        <li><a class="cite" href="https://doi.org/10.1006/tpbi.1998.1403">Wiuf
                and Hein (1999)</a> reformulated the CwR as a "spatial" process
                            along sequences.</li>
        <li><a class="cite"
        href="http://dx.doi.org/10.1098/rstb.2005.1673">McVean and Cardin
        (2005)</a> modified this to produce a process for local trees which is
    Markovian along sequences.</li>
    </ul>

    <div class="figure">
        <img data-src="smc_figure.svg" style="width:100%;padding:1em;">
        <div class="cite" style="font-size:0.8em;text-align:right">
            <a class="cite" href="http://dx.doi.org/10.1098/rstb.2005.1673">McVean and Cardin (2005)</a>
        </div>
    </div>

    <ul>
        <li>Sites between breakpoints exponentially distributed with rate $\rho T$ where $T$ is the total length of the current local tree.</li>
        <li>Neglects some possible recombinations, e.g. those that do not affect the data.</li>
    </ul>
</section>

<section>
    <h2>Population inference using SMC</h2>

    <ul>
        <li><a class="cite" href="http://dx.doi.org/10.1038/nature10231">Li &ampl; Durbin (2011)</a> developed an SMC-based HMM on pairs of alignments.</li>
        <li>Baum-Welch used to estimate parameters and hidden states (local TMRCAs).</li>
        <li>TMRCA distribution used to produce estimates of population size dynamics.</li>
    </ul>

    <div class="figure">
        <img data-src="li_durbin.jpg" style="height:400px;padding:1em;vertical-align:middle">
        <img data-src="li_durbin2.jpg" style="height:400px;padding:1em;vertical-align:bottom;margin-left:2em">
        <div class="cite" style="text-align:right;margin-top:1em"><a class="cite" href="http://dx.doi.org/10.1038/nature10231">Li and Durbin (2011)</a></div>
    </div>
</section>

<section>
    <h2>Bayesian inference of ARGs</h2>
</section>

<section>
    <h2>Summary</h2>
</section>

<section>
    <h2>Recommended Reading</h2>
</section>
