---
lecture_num: 16
---

<!-- 
Genome Assembly: An Introduction

    Sebastian Schmeier
    http://genomics.sschmeier.com
 -->

<section>
    <h2>Acknowledgement</h2>
    
    <div style="font-size:27px; margin-top:25%; text-align:middle">
    <p>Sebastian Schmeier</p>
    <p></p>
    <p>http://genomics.sschmeier.com</p>
    </div>
    
</section>

<section>
    <h2>Overview</h2>
        
    <ul class="spaced" style="margin-top:5%; width:85%">
        <li>Learning objectives</li>
        <li>DNA sequence assembly problem</li>
        <li>The greedy puzzle</li>
        <li>Graph-based genome assembly</li>
        <li>Seven bridges of Königsberg</li>
        <li>We construct a de Bruijn graph</li>
        <li>Underlying assumptions of graph-based genome assemblies</li>
        <li>Summary</li>
</li>
    </ul>
</section>

<section>
    <h2>Learning objectives</h2>


    <ul class="spaced" style="margin-top:5%; width:85%">
		<li> Be able to explain the DNA sequence assembly problem</li>
		<li> Be able to list the assumptions underlying a genome assembly</li>
		<li> Be able discuss how the assumptions relate to reality</li>
		<li> Be able to describe the principles of the greedy algorithm</li>
		<li> Be able to describe the principles underlying a graph-based genome assembly</li>
	</ul>

</section>

<section>
	<h2>DNA sequence assembly problem</h2>
	
<div style="margin-left:10%; margin-top:5%; width:50%; text-align:left">
    <ol class="spaced" style="margin-top:5%; width:50%">
	<li> The most commonly used DNA sequencing technology uses 
	random fragmentation and sequencing of short reads < 300nt</li>
	<li> This results in millions of short sequences</li>
	<li> Genome assembly involves joining overlapping short sequence 
	fragments together into long pieces to recover a continuous sequence of the genome</li>
	</ol>
	<b>How to do this efficiently?</b>
</div>
	
<div style="float:right; width:50%">
	<img data-src="assembly.svg" style="height:250px;">
</div>
</section>

<section>
	<h2>The greedy puzzle</h2>
	
<pre style="font-size:22px;box-shadow:none;margin-bottom:0">
<div style="float:left; width:5%; margin-left:15%;">
<font color="red">ATGCG</font>
<font color="green">GCGTG</font>
<font color="red">TGGCA</font>
<font color="green">GTGGC</font>
</div>
</pre>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Assumptions: 
	<ul class="spaced" style="margin-top:5%; width:75%">
	<li> If two sequence reads share a <i>common overlapping substring of letters</i>, 
	then it is because they are likely to have originated from the same chromosomal regions in the genome</li>
	</ul>
	</div>
</section>

<section>
	<h2>Objective : </h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Find a final string <i>“genome”</i> such that: 
	<ul class="spaced" style="margin-top:5%; width:75%">
	<li> all reads are substrings of <i>“genome”</i></li> 
	<li> <i>“genome”</i> should be as short as possible</li>
	</ul>
	</div>
	
<div style="float:left; width:5%; margin-left:15%;">
<pre style="font-size:22px;box-shadow:none;margin-bottom:0">
	<div style="float:left; width:5%; margin-left:15%;">
<s><font color="red">ATGCG</font></s>
<s><font color="green">GCGTG</font></s>    
<font color="organe">TGGCA</font>
<font color="blue">GTGGC</font>
	</div>
	
	<div style="float:middle; width:5%;">
	<font style="font-size:32px">&#8680;</font>
	</div>
	
	<div style="float:right; width:5%; margin-right:5%;">
<font color="red">ATGCG</font>
<font color="green">  GCGTG</font>    
	</div>
</pre>
</div>
	
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Greedy algorithm (find a solution incrementally): 
	<ol class="spaced" style="margin-top:5%; width:75%">
	<li> Pick the highest scoring overlap.</li> 
	<li> Merge the two overlapping fragments and add the resulting new sequence to the pool of sequences.</li>
	<li> Repeat until no more merges can be carried out.</li> 
	</ol>
	</div>

	
</section>

<section>
	<h2>Objective : </h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Find a final string <i>“genome”</i> such that: 
	<ul class="spaced" style="margin-top:5%; width:75%">
	<li> all reads are substrings of <i>“genome”</i></li> 
	<li> <i>“genome”</i> should be as short as possible</li>
	</ul>
	</div>
	
	<div style="text-align: left;">
	<div style="float:right; width:40%; margin-right:12%;">
<pre style="font-size:22px;box-shadow:none;margin-bottom:0">
<s><font color="red">ATGCG</font></s>
<s><font color="green">GCGTG</font></s>    
<font color="organe">TGGCA</font>
<s><font color="blue">GTGGC</font></s>
</pre>
	</div>
	
	&#8680;
	
	<div style="float:right; width:40%; margin-right:12%;">
<pre style="font-size:22px;box-shadow:none;margin-bottom:0">
<font color="red">ATGCG</font>
<font color="green">  GCGTG</font>    
<font color="blue">    GTGGC</font>
</pre>
	
	</div>
	</div>
	
		<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Greedy algorithm (find a solution incrementally): 
	<ol class="spaced" style="margin-top:5%; width:75%">
	<li> Pick the highest scoring overlap.</li> 
	<li> Merge the two overlapping fragments and add the resulting new sequence to the pool of sequences.</li>
	<li> Repeat until no more merges can be carried out.</li> 
	</ol>
	</div>

	
</section>


<section>
	<h2>Objective : </h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Find a final string <i>“genome”</i> such that: 
	<ul class="spaced" style="margin-top:5%; width:75%">
	<li> all reads are substrings of <i>“genome”</i></li> 
	<li> <i>“genome”</i> should be as short as possible</li>
	</ul>
	</div>
	
	<div style="float:right; width:40%; margin-right:12%;">
<pre style="font-size:22px;box-shadow:none;margin-bottom:0">
<s><font color="red">ATGCG</font></s>
<s><font color="green">GCGTG</font></s>    
<s><font color="organe">TGGCA</font></s> 
<s><font color="blue">GTGGC</font></s> 
</pre>
	</div>
	
	&#8680;
	
	<div style="float:right; width:40%; margin-right:12%;">
<pre style="font-size:22px;box-shadow:none;margin-bottom:0">
<font color="red">ATGCG</font>
<font color="grey">|</font><font color="green"> GCGTG</font>    
<font color="grey">|</font><font color="blue">   GTGGC</font>
<font color="grey">|</font><font color="organe">    TGGCA</font>
<font color="grey">|        |</font>
<b><font color="grey">ATGCGTGGCA</font></b>
<font style="font-size:12px">           genome</font>
</pre>
	
	</div>
	
	
		<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Greedy algorithm (find a solution incrementally): 
	<ol class="spaced" style="margin-top:5%; width:75%">
	<li> Pick the highest scoring overlap.</li> 
	<li> Merge the two overlapping fragments and add the resulting new sequence to the pool of sequences.</li>
	<li> Repeat until no more merges can be carried out.</li> 
	</ol>
	</div>

	
</section>


<section>
	<h2>Greedy algorithms are not a good solution : </h2>
		
	<div style="float:right; width:40%; margin-right:12%;">
<pre style="font-size:22px;box-shadow:none;margin-bottom:0">
<s><font color="red">ATGCG</font></s>
<s><font color="green">GCGTG</font></s>    
<s><font color="organe">TGGCA</font></s> 
<s><font color="blue">GTGGC</font></s> 
</pre>
	</div>
	
	&#8680;
	
	<div style="float:right; width:40%; margin-right:12%;">
<pre style="font-size:22px;box-shadow:none;margin-bottom:0">
<font color="red">ATGCG</font>
<font color="grey">|</font><font color="green"> GCGTG</font>    
<font color="grey">|</font><font color="blue">   GTGGC</font>
<font color="grey">|</font><font color="organe">    TGGCA</font>
<font color="grey">|        |</font>
<b><font color="grey">ATGCGTGGCA</font></b>
<font style="font-size:12px">           genome</font>
</pre>
	
	</div>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Assumptions: 
	<ul class="spaced" style="margin-top:5%; width:75%">
	<li> Reads are 100% accurate</li> 
	<li> Identical reads must come from the same location on the genome</li>
	<li> All substrings from the genome are represented in the set of sequenced reads</li> 
	</ul>
	<b><font color="red">But none of them hold true</font></b>
	</div>

</section>


<section>
	<h2>Graph theory</h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Represent relationships between entities 
	</div>
	
	<div style="float:middle; width:50%">
	<img data-src="graph.svg">
	</div>
</section>

<section>
	<h2>Graph theory</h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Analyse connections e.g. find well–connected hubs in graphs
	</div>
	
	<div style="float:middle; width:50%">
	<img data-src="graph2.svg">
	</div>
</section>

<section>
	<h2>Graph theory</h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Directed graph e.g. information flow 
	</div>
	
	<div style="float:middle; width:50%">
	<img data-src="graph3.svg">
	</div>	
</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly1.svg" style="width:100%">
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Preprocess the sequence reads to determine the pair-wise overlap information and represent these binary relationships 
	</div>
</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly2.svg" style="width:100%">
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Preprocess the sequence reads to determine the pair-wise overlap information and represent these binary relationships
	</div>
</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly3.svg" style="width:100%">
	
	<ul class="spaced" style="margin-top:5%; width:75%">
	<li> Preprocess the sequence reads to determine the pair-wise overlap information 
	and represent these binary relationships</li> 
	<li> The problem of finding a consistent lay-out (i.e. genome) 
	can then be formulated in terms of searching a paths in the graph satisfying certain criteria</li> 
	</ul>

</section>

<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="graph-assembly4.svg" style="width:100%">
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Assumption:  All substrings are represented
	<ul class="spaced" style="margin-top:5%; width:75%">
	<li> Even modern sequencers that generate 150nt+ reads do not cover all possible 150-mers present in the genome</li> 
	<li> Thus, people generally use substrings of certain length k (k-mers)</li> 
	</ul>
    &rarr; Here we use 3-mers by cutting the original reads into reads of length 3
    </div>

</section>


<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="graph-assembly5.svg" style="width:100%">
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Make them unique
    </div>
</section>


<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="graph-assembly6.svg" style="width:100%">
	
	<ul class="spaced" style="margin-top:5%; width:85%">
	<li> Each k-mer is a vertex</li>
	<li> Draw edge from x to y, where		</li> 
		<ul class="spaced" style="margin-top:5%; width:75%">
		<li> <u>suffix</u> from x overlaps <u>prefix</u> from y</li> 
		</ul> 
	</ul>

</section>


<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="graph-assembly7.svg" style="width:100%">
		
	<ul class="spaced" style="margin-top:5%; width:85%">
	<li> Each k-mer is a vertex</li>
	<li> Draw edge from x to y, where		</li> 
		<ul class="spaced" style="margin-top:5%; width:75%">
		<li> <u>suffix</u> from x overlaps <u>prefix</u> from y</li> 
		</ul> 
	</ul>

</section>

<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="graph-assembly8.svg" style="width:100%">
		
	<ul class="spaced" style="margin-top:5%; width:85%">
	<li> Each k-mer is a vertex</li>
	<li> Draw edge from x to y, where		</li> 
		<ul class="spaced" style="margin-top:5%; width:75%">
		<li> <u>suffix</u> from x overlaps <u>prefix</u> from y</li> 
		</ul> 
	</ul>

</section>


<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="graph-assembly9.svg" style="width:100%">
		
	<ul class="spaced" style="margin-top:5%; width:85%">
	<li> Each k-mer is a vertex</li>
	<li> Draw edge from x to y, where		</li> 
		<ul class="spaced" style="margin-top:5%; width:75%">
		<li> <u>suffix</u> from x overlaps <u>prefix</u> from y</li> 
		</ul> 
	</ul>

</section>


<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="hamiltonian-path1.svg" style="width:100%">
		
	<ul class="spaced" style="margin-top:5%; width:85%">
	<li> Find <b><u>Hamiltonian path</u></b>, that is, a path that visits every <u>vertex</u> exactly once</li>
	<li> Record the First letter of each vertex + All letters of last vertex</li> 
	</ul>

</section>

<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="hamiltonian-path2.svg" style="width:100%">
		
	<ul class="spaced" style="margin-top:5%; width:85%">
	<li> Find <b><u>Hamiltonian path</u></b>, that is, a path that visits every <u>vertex</u> exactly once</li>
	<li> Record the First letter of each vertex + All letters of last vertex</li> 
	</ul>

    <div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	<b>UNFORTUNATELY:</b> <br>
	&nbsp; &nbsp; &rarr; The Hamiltonian path problem is very difficult to solve (NP-complete)
    </div>
    
</section>

<section>
<!-- 	<h2>Graph-based genome assembly</h2> -->
		
	<img data-src="hamiltonian-path3.svg" style="width:100%">
		
	<ul class="spaced" style="margin-top:5%; width:85%">
	<li> Find <b><u>Hamiltonian path</u></b>, that is, a path that visits every <u>vertex</u> exactly once</li>
	<li> Record the First letter of each vertex + All letters of last vertex</li> 
	</ul>

    <div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	<b>UNFORTUNATELY:</b> <br>
	&nbsp; &nbsp; &rarr; The Hamiltonian path problem is very difficult to solve (NP-complete)
    </div>

</section>


<section>
	<h2>Seven bridges of Königsberg </h2>

    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> In 1735 Leonhard Euler was presented with the following problem:</li>
		<ul class="spaced" style="margin-top:5%; width:75%">
		<li> Find a walk through the city that would cross each bridge once and only once</li> 
		<li> He proved that a such a walk does not exist for Königsberg </li> 
		</ul> 
	<li> Computationally finding an Eulerian path is much easier than an Hamiltonian</li> 
	</ul>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	&rarr; <u> we need to reformulate our assembly problem </u> 
    </div>

	<img data-src="SevenBridges.svg" style="width:100%">
	
</section>












