---
lecture_num: 16
---

<!-- 
Genome Assembly: An Introduction

    Sebastian Schmeier
    http://genomics.sschmeier.com
 -->

<section>
    <h2>Acknowledgement</h2>
    
    <div style="font-size:27px; margin-top:25%; margin-left:25%; text-align:left">
    <p>Sebastian Schmeier</p>
    <p>s.schmeier@gmail.com</p>
    <p>http://genomics.sschmeier.com</p>
    </div>
    
</section>

<section>
    <h2>Overview</h2>
        
    <ul class="spaced" style="margin-top:5%; width:85%">
        <li>Learning objectives</li>
        <li>DNA sequence assembly problem</li>
        <li>The greedy puzzle</li>
        <li>Graph-based genome assembly</li>
        <li>Seven bridges of Königsberg</li>
        <li>We construct a de Bruijn graph</li>
        <li>Underlying assumptions of graph-based genome assemblies</li>
        <li>Summary</li>
    </ul>
</section>

<section>
    <h2>Learning objectives</h2>

    <ul class="spaced" style="margin-top:5%; width:85%">
		<li> Be able to explain the DNA sequence assembly problem</li>
		<li> Be able to list the assumptions underlying a genome assembly</li>
		<li> Be able discuss how the assumptions relate to reality</li>
		<li> Be able to describe the principles of the greedy algorithm</li>
		<li> Be able to describe the principles underlying a graph-based genome assembly</li>
	</ul>

</section>

<section>
	<h2>DNA sequence assembly problem</h2>
	
	<div>
<div style="margin-left:5%; margin-top:5%; width:50%">
    <ol class="spaced"">
	<li> The most commonly used DNA sequencing technology uses 
	random fragmentation and sequencing of short reads < 300nt</li>
	<li> This results in millions of short sequences</li>
	<li> Genome assembly involves joining overlapping short sequence 
	fragments together into long pieces to recover a continuous sequence of the genome</li>
	<li style="list-style-type:none"><b>How to do this efficiently?</b></li>
    </ol>
</div>
	
<div style="float:right; width:30%">
	<img data-src="assembly.svg" style="height:60%">
</div>
</div>

</section>


<section>
	<h2>The greedy puzzle</h2>
	
	<div style="width:5%; margin-left:15%; width:90%">
	<pre style="font-size:30px;box-shadow:none">
	<font color="red">ATGCG</font>
	<font color="green">GCGTG</font>
	<font color="red">TGGCA</font>
	<font color="green">GTGGC</font>
	</pre>
	</div>
	
	<div align="left" style="margin-left:15%; width:85%">
	Assumptions: 
	<ul class="spaced" style="margin-top:5%">
	<li> If two sequence reads share a <i>common overlapping substring of letters</i>, 
	then it is because they are likely to have originated from the same chromosomal regions in the genome</li>
	</ul>
	</div>
</section>

<section>
	<h2>Objective : </h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Find a final string <i>“genome”</i> such that: 
	<ul class="spaced" style="margin-top:5%">
	<li> all reads are substrings of <i>“genome”</i></li> 
	<li> <i>“genome”</i> should be as short as possible</li>
	</ul>
	</div>
	
	<img data-src="greedy-alg2.svg" style="width:60%">
	
	<div align="left" style="margin-left:15%; width:85%">
	Greedy algorithm (find a solution incrementally): 
	<ol class="spaced" style="margin-top:5%">
	<li> Pick the highest scoring overlap.</li> 
	<li> Merge the two overlapping fragments and add the resulting new sequence to the pool of sequences.</li>
	<li> Repeat until no more merges can be carried out.</li> 
	</ol>
	</div>
</section>

<section>
	<h2>Objective : </h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Find a final string <i>“genome”</i> such that: 
	<ul class="spaced" style="margin-top:5%">
	<li> all reads are substrings of <i>“genome”</i></li> 
	<li> <i>“genome”</i> should be as short as possible</li>
	</ul>
	</div>
	
	<img data-src="greedy-alg3.svg" style="width:60%">
	
	<div align="left" style="margin-left:15%; width:85%">
	Greedy algorithm (find a solution incrementally): 
	<ol class="spaced" style="margin-top:5%">
	<li> Pick the highest scoring overlap.</li> 
	<li> Merge the two overlapping fragments and add the resulting new sequence to the pool of sequences.</li>
	<li> Repeat until no more merges can be carried out.</li> 
	</ol>
	</div>
</section>


<section>
	<h2>Objective : </h2>
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Find a final string <i>“genome”</i> such that: 
	<ul class="spaced" style="margin-top:5%">
	<li> all reads are substrings of <i>“genome”</i></li> 
	<li> <i>“genome”</i> should be as short as possible</li>
	</ul>
	</div>
	
	<img data-src="greedy-alg-cross.svg" style="width:60%">
	
	<div align="left" style="margin-left:15%; width:85%">
	Greedy algorithm (find a solution incrementally): 
	<ol class="spaced" style="margin-top:5%">
	<li> Pick the highest scoring overlap.</li> 
	<li> Merge the two overlapping fragments and add the resulting new sequence to the pool of sequences.</li>
	<li> Repeat until no more merges can be carried out.</li> 
	</ol>
	</div>
</section>


<section>
	<h2>Greedy algorithms are not a good solution : </h2>
		
	<img data-src="greedy-alg-cross.svg" style="width:60%">
	
	<div align="left" style="margin-left:15%; width:85%">
	Assumptions: 
	<ul class="spaced" style="margin-top:5%">
	<li> Reads are 100% accurate</li> 
	<li> Identical reads must come from the same location on the genome</li>
	<li> All substrings from the genome are represented in the set of sequenced reads</li> 
	<li style="list-style-type:none"> <b><font color="red">But none of them hold true</font></b></li> 
	</ul>
	</div>

</section>

<section>
	<h2>Greedy algorithms are not a good solution : </h2>
	
	<ul class="spaced" style="margin-top:5%; width:85%">
	<li> Greedy algorithm for genome assembly are extremely slow.</li> 
	<li> It is not feasible to do this iterative process with millions of reads in 
	a reasonable time and expect an “optimal” solution.</li> 
	</ul>
	
	<img data-src="greedy-alg.svg" style="width:60%">
	
</section>



<section>
	<h2>Graph theory</h2>
	
	<div align="left" style="margin-top:5%; margin-left:5%; width:85%">
	Represent relationships between entities 
	</div>
	
	<div style="float:left; margin-left:25%; width:60%">
	<img data-src="graph.svg">
	</div>
</section>

<section>
	<h2>Graph theory</h2>
	
	<div align="left" style="margin-top:5%; margin-left:5%; width:85%">
	Analyse connections e.g. find well–connected hubs in graphs
	</div>
	
	<div style="float:left; margin-left:25%; width:60%">
	<img data-src="graph2.svg">
	</div>
</section>

<section>
	<h2>Graph theory</h2>
	
	<div align="left" style="margin-top:5%; margin-left:5%; width:85%">
	Directed graph e.g. information flow 
	</div>
	
	<div style="float:left; margin-left:25%; width:60%">
	<img data-src="graph3.svg">
	</div>	
</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly1.svg" style="width:80%">
	
	<div align="left" style="margin-left:15%; width:85%">
	Preprocess the sequence reads to determine the pair-wise overlap information and represent these binary relationships 
	</div>
</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly2.svg" style="width:80%">
	
	<div align="left" style="margin-left:15%; width:85%">
	Preprocess the sequence reads to determine the pair-wise overlap information and represent these binary relationships
	</div>
</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly3.svg" style="width:80%">
	
	<ul class="spaced" style="width:85%">
	<li> Preprocess the sequence reads to determine the pair-wise overlap information 
	and represent these binary relationships</li> 
	<li> The problem of finding a consistent lay-out (i.e. genome) 
	can then be formulated in terms of searching a paths in the graph satisfying certain criteria</li> 
	</ul>

</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly4.svg" style="width:80%">
	
	<ul class="spaced" style="width:85%">
	<li style="list-style-type:none"> <u>Assumption: all substrings are represented</u></li>
		<ul>
		<li> Even modern sequencers that generate 150nt+ reads <b><u>do not</u></b> 
		cover all possible 150-mers present in the genome</li> 
		<li> Thus, people generally use substrings of certain length k (k-mers)</li> 
		</ul>
    <li style="list-style-type:none"> &rarr; Here we use 3-mers by cutting the original reads into reads of length 3</li> 
    </ul>
</section>


<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly5.svg" style="width:80%">
	
	<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	Make them unique
    </div>
</section>


<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly6.svg" style="width:80%">
	
	<ul class="spaced" style="width:85%">
	<li> Each k-mer is a vertex</li>
	<li> Draw edge from x to y, where		</li> 
		<ul>
		<li> <u>suffix</u> from x overlaps <u>prefix</u> from y</li> 
		</ul> 
	</ul>

</section>


<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly7.svg" style="width:80%">
		
	<ul class="spaced" style="width:85%">
	<li> Each k-mer is a vertex</li>
	<li> Draw edge from x to y, where		</li> 
		<ul>
		<li> <u>suffix</u> from x overlaps <u>prefix</u> from y</li> 
		</ul> 
	</ul>

</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly8.svg" style="width:80%">
		
	<ul class="spaced" style="width:85%">
	<li> Each k-mer is a vertex</li>
	<li> Draw edge from x to y, where		</li> 
		<ul>
		<li> <u>suffix</u> from x overlaps <u>prefix</u> from y</li> 
		</ul> 
	</ul>

</section>


<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="graph-assembly9.svg" style="width:80%">
		
	<ul class="spaced" style="width:85%">
	<li> Each k-mer is a vertex</li>
	<li> Draw edge from x to y, where		</li> 
		<ul>
		<li> <u>suffix</u> from x overlaps <u>prefix</u> from y</li> 
		</ul> 
	</ul>

</section>


<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="hamiltonian-path1.svg" style="width:80%">
		
	<ul class="spaced" style="width:85%">
	<li> Find <b><u>Hamiltonian path</u></b>, that is, a path that visits every <u>vertex</u> exactly once</li>
	<li> Record the First letter of each vertex + All letters of last vertex</li> 
	</ul>

</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="hamiltonian-path2.svg" style="width:80%">
		
	<ul class="spaced" style="width:85%">
	<li> Find <b><u>Hamiltonian path</u></b>, that is, a path that visits every <u>vertex</u> exactly once</li>
	<li> Record the First letter of each vertex + All letters of last vertex</li> 
	<li style="list-style-type:none"><b>UNFORTUNATELY:</b></li> 
	     <ul class="spaced">
	<li style="list-style-type:none">&rarr; The Hamiltonian path problem is very difficult to solve (NP-complete)</li> 
         </ul>
    </ul>
</section>

<section>
	<h2>Graph-based genome assembly</h2>
		
	<img data-src="hamiltonian-path3.svg" style="width:80%">
		
	<ul class="spaced" style="width:85%">
	<li> Find <b><u>Hamiltonian path</u></b>, that is, a path that visits every <u>vertex</u> exactly once</li>
	<li> Record the First letter of each vertex + All letters of last vertex</li> 
	<li style="list-style-type:none"><b>UNFORTUNATELY:</b></li> 
	     <ul class="spaced">
	<li style="list-style-type:none">&rarr; The Hamiltonian path problem is very difficult to solve (NP-complete)</li> 
         </ul>
    </ul>

</section>


<section>
	<h2>Seven bridges of Königsberg </h2>

    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> In 1735 Leonhard Euler was presented with the following problem:</li>
		<ul class="spaced" style="width:75%">
		<li> Find a walk through the city that would cross each bridge once and only once</li> 
		<li> He proved that a such a walk does not exist for Königsberg </li> 
		</ul> 
	<li> Computationally finding an Eulerian path is much easier than an Hamiltonian</li> 
	</ul>
	
	<div align="left" style="margin-left:15%; width:85%">
	&rarr; <u> we need to reformulate our assembly problem </u> 
    </div>

	<img data-src="SevenBridges.svg" style="width:100%">
	
</section>


<section>
	<h2>de Bruijn graph</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
    <p>We construct a <b> de Bruijn graph :</b> </p>
    <ul class="spaced" style="width:85%">
	<li> edges represent k-mers </li>
	<li> vertices correspond to (k-1)-mers</li> 
	</ul>
	<ol class="spaced" style="width:75%">
	<li> Form a node for every distinct prefix or suffix of a k-mer</li> 
	<li> Connect vertex x to vertex y with a directed edge if some k-mer 
	(e.g., ATG) has prefix x (e.g., AT) and suffix y (e.g., TG), and label the edge with this k-mer. </li> 
	</ol> 
    <br>
    Original reads:
<font color="red">ATGCG</font>&nbsp; &nbsp; 
<font color="green">GCGTG</font>&nbsp; &nbsp; 
<font color="blue">GTGGC</font>&nbsp; &nbsp; 
<font color="orange">TGGCA</font>&nbsp; &nbsp; 

	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT</p>
    <p><b>Distinct (k-1)-mers: </b> </p>

    </div>
	
</section>


<section>
	<h2>de Bruijn graph</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
    <p>We construct a <b> de Bruijn graph :</b> </p>
    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> edges represent k-mers </li>
	<li> vertices correspond to (k-1)-mers</li> 
	</ul>
	<ol class="spaced" style="margin-top:5%; width:75%">
	<li> Form a node for every distinct prefix or suffix of a k-mer</li> 
	<li> Connect vertex x to vertex y with a directed edge if some k-mer 
	(e.g., ATG) has prefix x (e.g., AT) and suffix y (e.g., TG), and label the edge with this k-mer. </li> 
	</ol> 
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;<font color="red"><u>AT</u></font>G, TGG, TGC, GTG, GGC, GCA, GCG, CGT</p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<div align="left" style="margin-left:15%; width:85%">
<svg height="350" width="350">
    <circle cx="50" cy="50" r="25"
            stroke="black"
            stroke-width="3"
            fill="none"
            />
    <text x="50" y="55" 
          text-anchor="middle"
          stroke="black"
          stroke-width="1px"
          > AT
    </text>
</svg>
</div>
    
</section>

<section>
	<h2>de Bruijn graph</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
    <p>We construct a <b> de Bruijn graph :</b> </p>
    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> edges represent k-mers </li>
	<li> vertices correspond to (k-1)-mers</li> 
	</ul>
	<ol class="spaced" style="margin-top:5%; width:75%">
	<li> Form a node for every distinct prefix or suffix of a k-mer</li> 
	<li> Connect vertex x to vertex y with a directed edge if some k-mer 
	(e.g., ATG) has prefix x (e.g., AT) and suffix y (e.g., TG), and label the edge with this k-mer. </li> 
	</ol> 
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;A<font color="red"><u>TG</u></font>, <font color="red"><u>TG</u></font>G, 
	<font color="red"><u>TG</u></font>C, G<font color="red"><u>TG</u></font>, GGC, GCA, GCG, CGT</p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<div align="left" style="margin-left:15%; width:85%">
<svg height="350" width="350">
    <circle cx="50" cy="50" r="25"
            stroke="black"
            stroke-width="3"
            fill="none"
            />
    <text x="50" y="55" 
          text-anchor="middle"
          stroke="black"
          stroke-width="1px"
          > AT
    </text>
    <circle cx="150" cy="50" r="25"
            stroke="black"
            stroke-width="3"
            fill="none"
            />
    <text x="150" y="55" 
          text-anchor="middle"
          stroke="black"
          stroke-width="1px"
          > TG
    </text>

</svg>
</div>
    
</section>


<section>
	<h2>de Bruijn graph</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
    <p>We construct a <b> de Bruijn graph :</b></p> 
    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> edges represent k-mers </li>
	<li> vertices correspond to (k-1)-mers</li> 
	</ul>
	<ol class="spaced" style="margin-top:5%; width:75%">
	<li> Form a node for every distinct prefix or suffix of a k-mer</li> 
	<li> Connect vertex x to vertex y with a directed edge if some k-mer 
	(e.g., ATG) has prefix x (e.g., AT) and suffix y (e.g., TG), and label the edge with this k-mer. </li> 
	</ol> 
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, T<font color="red"><u>GG</u></font>, 
	TGC, GTG, <font color="red"><u>GG</u></font>C, GCA, GCG, CGT</p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<div align="left" style="margin-left:15%; width:85%">
<svg height="350" width="350">
    <circle cx="50" cy="50" r="25"
            stroke="black"
            stroke-width="3"
            fill="none"
            />
    <text x="50" y="55" 
          text-anchor="middle"
          stroke="black"
          stroke-width="1px"
          > AT
    </text>
    <circle cx="150" cy="50" r="25"
            stroke="black"
            stroke-width="3"
            fill="none"
            />
    <text x="150" y="55" 
          text-anchor="middle"
          stroke="black"
          stroke-width="1px"
          > TG
    </text>
    <circle cx="250" cy="100" r="25"
            stroke="black"
            stroke-width="3"
            fill="none"
            />
    <text x="250" y="105" 
          text-anchor="middle"
          stroke="black"
          stroke-width="1px"
          > GG
    </text>
</svg>
</div>
    
</section>

<section>
	<h2>de Bruijn graph</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT </p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<img data-src="deBruijnGraph.svg" style="width:75%">
    
</section>

<section>
	<h2>de Bruijn graph</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT </p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<img data-src="deBruijnGraph2.svg" style="width:75%">
    
</section>

<section>
	<h2>de Bruijn graph</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT </p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<img data-src="deBruijnGraph3.svg" style="width:75%">
    
</section>

<section>
	<h2>de Bruijn graph</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT </p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<img data-src="deBruijnGraph4.svg" style="width:75%">
    
</section>


<section>
	<h2>Find a DNA sequence containing all k-mers?</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> In a de Bruijn graph, can we find a path that visits every edge of the graph exactly once? </li>
	<li> Eulerian path</li> 
	</ul>
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT </p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<img data-src="deBruijnGraph4.svg" style="width:75%">
    
</section>

<section>
	<h2>Find a DNA sequence containing all k-mers?</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> In a de Bruijn graph, can we find a path that visits every edge of the graph exactly once? </li>
	<li> Eulerian path</li> 
	</ul>
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT </p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<img data-src="deBruijnGraph5.svg" style="width:75%">
    
</section>

<section>
	<h2>Find a DNA sequence containing all k-mers?</h2>

<div align="left" style="margin-top:5%; margin-left:15%; width:85%">
    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> In a de Bruijn graph, can we find a path that visits every edge of the graph exactly once? </li>
	<li> Eulerian path</li> 
	</ul>
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT </p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<img data-src="deBruijnGraph6.svg" style="width:75%">
    
</section>

<section>
	<h2>Find a DNA sequence containing all k-mers?</h2>

<div align="left" style="margin-top:5%; width:85%">
    <ul class="spaced" style="margin-top:5%; width:85%">
	<li> In a de Bruijn graph, can we find a path that visits every edge of the graph exactly once? </li>
	<li> Eulerian path</li> 
	</ul>
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, TGG, TGC, GTG, GGC, GCA, GCG, CGT </p>
    <p><b>Distinct (k-1)-mers: </b> </p>
</div>

<img data-src="deBruijnGraph7.svg" style="width:85%">
    
</section>


<section>
    <h2>More assumptions of graph-based genome assemblies</h2>

    <ul class="spaced" style="margin-top:5%; width:85%">
		<li> Hidden assumptions that do <b>not</b> hold for short read sequencing. 
		We took for granted that:</li>
		<ol>
			<li> We can generate all k-mers present in the genome</li>
			<li> All k-mers are error free</li>
			<li> Each k-mer appears at most once in the genome </li>
		</ol>
	</ul>

</section>


<section>
    <h2>More assumptions of graph-based genome assemblies</h2>

    <ul class="spaced" style="width:85%">
		<li> Hidden assumptions that do <b>not</b> hold for short read sequencing. 
		We took for granted that:</li>
		<ol>
		<li> We can generate all k-mers present in the genome</li>
		<li> <b>All k-mers are error free</b></li>
		<li> Each k-mer appears at most once in the genome </li>
		</ol>
		
		<li> Due to these reasons we do <b>NOT</b> choose the longest possible k-mer</li>
		<li> The <b>smaller</b> the k-mer the higher the probability that we see <b>all</b> k-mers</li>
		<li> Also, errors:</li>
	</ul>

<img data-src="kmers.svg" style="width:85%">

</section>

<section>
    <h2>More assumptions of graph-based genome assemblies</h2>

    <ul class="spaced" style="margin-top:5%; width:85%">
		<li> Hidden assumptions that do <b>not</b> hold for short read sequencing. 
		We took for granted that:</li>
		<ol>
		<li> We can generate all k-mers present in the genome</li>
		<li> All k-mers are error free</li>
		<li> <b>Each k-mer appears at most once in the genome</b> </li>
		<ul>
			<li style="list-style-type:none"> <b>&rarr; Repeats </b> </li>
			<li> This is most often not true</li>
			<li> This is known as k-mer multiplicity</li>
		</ul>
		</ol>
	</ul>

</section>

<section>
    <h2>More assumptions of graph-based genome assemblies</h2>

<div align="left" style="margin-left:15%; width:85%">
	<p>k-mers:&nbsp;&nbsp;&nbsp;&nbsp;ATG, GCA, TGC, <font color="green">TGC</font>, GTG, 
	             <font color="green">GTG</font>, GCG, <font color="green">GCG</font>, CGT, CGT </p>
    <p><b>Distinct (k-1)-mers:</b>&nbsp;&nbsp;&nbsp;&nbsp;A<font color="red">TGCG</font><font color="orange">TGCG</font>TGCA </p>
</div>

<img data-src="kmers-graph.svg" style="width:85%">

</section>


<section>
    <h2>Summary</h2>

    <ul class="spaced" style="margin-top:5%; width:85%">
		<li> DNA sequence assembly problem</li>
		<ul>
		<li> Given millions of short sequences, join overlapping short sequence fragments 
		into long pieces to recover a continuous sequence of the genome.</li>
		</ul>
		<li> Greedy assembly</li>
		<ul>
		<li> Find the shortest common superstring by iteratively joining and merging overlapping reads until all reads have been merged.</li>
		</ul>
		<li> Graph-based assembly</li>
		<ul>
		<li> Pre-process sequence reads to determine the pair-wise overlap information of substrings and represent these binary relationships in a graph.</li>
		<li> Search a consistent genome layout by finding an eulerian path (visit each edge once) through the graph satisfying certain criteria.</li>
		</ul>		
	</ul>
</section>

<section>
    <h2>Summary (2)</h2>

    <ul class="spaced" style="margin-top:5%; width:95%">
		<li> Assumptions:</li>
		<ul>
			<li> If two sequence reads share a common overlapping substring, 
			they are likely originating from the same chromosomal regions in the genome</li>
			<li> Sequenced reads are 100% accurate (not true)</li>
			<li> Identical reads must come from the same location on the genome (mostly not true)</li>
			<li> All substrings from the genome are represented in the set of sequenced reads (very likely not true)</li>
			<li> All k-mers are error free (not true)</li>
			<li> We can generate all k-mers present in the genome (very likely untrue)</li>
			<li> Each k-mer appears at most once in the genome (likely untrue)</li>
		</ul>		
	</ul>
	
	<p style="font-size:10px">
	References: <br>
How to apply de Bruijn graphs to genome assembly. Phillip E C Compeau, Pavel A Pevzner & Glenn Tesler (2011) . 
Nature Biotechnology 29, 987–991 <br>
Sequence Assembly. Lecture by Mark Craven (craven@biostat.wisc.edu). BMI/CS 576 (www.biostat.wisc.edu/bmi576/), Fall 2011<br>	
Comparing De Novo Genome Assembly: The Long and Short of It. Narzisi G, Mishra B (2011) PLoS ONE6(4): e19175.<br> 
</p>

</section>

<section>
    <h2>Seven bridges of Königsberg II</h2>

    <ul class="spaced" style="margin-top:5%; width:85%">
		<li> The case of <u>directed graphs</u> is similar:</li>
		<ul>
		<li> A graph in which indegrees are equal to outdegrees for all nodes is called 'balanced'. </li>
		<li> Euler's theorem states that <u>a connected directed graph has an Eulerian cycle if and only if it is balanced</u>. </li>
		</ul>		
	</ul>

</section>

<section>
    <h2>References</h2>

    <ul class="spaced" style="margin-top:5%; width:85%">
		<li> How to apply de Bruijn graphs to genome assembly. Phillip E C Compeau, Pavel A Pevzner & Glenn Tesler. 
		Nature Biotechnology 29, 987–991 (2011) doi:10.1038/nbt.2023 Published online 08 November 2011 </li>
		<li> Sequence Assembly. Lecture by Mark Craven (craven@biostat.wisc.edu). BMI/CS 576 (www.biostat.wisc.edu/bmi576/), Fall 2011 </li>
		<li> Comparing De Novo Genome Assembly: The Long and Short of It. Narzisi G, Mishra B (2011) PLoS ONE6(4): e19175.</li>	
	</ul>
    <p>Sebastian Schmeier</p>
    <p>s.schmeier@gmail.com</p>
    <p>http://genomics.sschmeier.com</p>
</section>


