---
lecture_num: 13
---

<section>
    <h2>Master equations</h2>

    <ul>
        <li>Describe probability distribution of continuous-time "jump" processes.</li>
        <li>Form can be derived directly from the Chapman-Kolmogorov equation:
            $$p(x_3,t_3|x_1,t_1) = \sum_{x_2}p(x_3,t_3|x_2,t_2)p(x_2,t_2|x_1,t_1)$$
            for any $t_3&gt;t_2&gt;t_1$, where $p(x_1,t_1)\equiv P(X(t_1)=x_1)$.
        </li>
    </ul>

    <p>Derivation:</p>
    <ol>
        <li>$\partial_t p(x,t|x_0) = \lim_{\Delta\rightarrow 0}\frac{1}{\Delta}\left(p(x,t+\Delta|x_0)-p(x,t|x_0)\right)$ <b>(Definition)</b></li>
        <li>$p(x,t+\Delta|x_0)=\sum_{x'}p(x,t+\Delta|x',t)p(x',t|x_0)$ <b>(CKE)</b></li>
        <li>$p(x,t+\Delta|x',t) \rightarrow \delta_{x,x'} + \Delta W(x|x')$ for small $\Delta$ <b>(Assumption)</b></li>
        <li>Then $p(x,t+\Delta|x_0) = p(x,t|x_0) + \Delta\sum_{x'}W(x|x')p(x',t|x_0)$</li>
        <li>and $\partial_t p(x,t|x_0) = \sum_{x'}W(x|x')p(x',t|x_0)$.</li>
    </ol>
    <blockquote style="font-size=0.8em;width:100%">
        $\partial_t p(x,t|x_0) = \sum_{x'\neq x}W(x|x')p(x',t|x_0) - \sum_{x'\neq x}W(x'|x)p(x,t|x_0)$
    </blockquote>

    For $x'\neq x$, the quantity $\Delta W(x'|x)$ is the probability of the system transitioning from state $x$ to state $x'$ in the time interval $\Delta$. (The <b>transition rate</b>.)
</section>

<section>
    <h2>Birth-death master equations</h2>

    <p>Suppose $\vec{X}(t)$ describes the state of a system composed of distinct sub-populations $i$ each containing $X_i(t)$ individuals evolving under a continuous-time Markov process. Call this a birth-death process if these counts can only increase or decrease by finite amounts in a given event.</p>

    <p>
    $$W(\vec{x}'|\vec{x})=
    \delta_{\vec{x}',\vec{x}+\vec{v}_1}T_1(\vec{x})
    + \delta_{\vec{x}',\vec{x}+\vec{v}_2}T_2(\vec{x})
    + \ldots$$
    where
    <ul>
        <li>$T_k(\vec{x})$ is the probability per unit time of event $k$ occuring given $\vec{X}(t)=\vec{x}$
            and</li>
        <li> $\vec{v}_k$ is the increment applied to the system state by event $k$.</li>
    </ul></p>

    <p>Then,
    $$\partial_t p(\vec{x},t|\vec{x}_0)=\sum_k\left[T_k(\vec{x}-\vec{v}_k)p(\vec{x}-\vec{v}_k|\vec{x}_0)
    - T_k(\vec{x})p(\vec{x},t|\vec{x}_0)\right]$$
    and the mean dynamics are
    $$\partial_t\langle \vec{X}(t)\rangle = \sum_k \vec{v}_k\langle T_k(\vec{X}(t))\rangle
    \simeq \sum_k\vec{v}_k T_k(\langle\vec{X}(t)\rangle)$$
</section>

<section>
    <h2>Chemical master equations</h2>

    <p>Consider the chemical reaction
    $$r_1 A_1 + r_2 A_2 + \ldots \overset{\lambda}{\longrightarrow} p_1 A_1 + p_2 A_2 + \ldots$$
    where $r_i$ and $p_i$ specify the (integer) number of $A_i$ particles involved on the reactant and product sides, respectively.</p>

    <p>Assuming the reaction is occuring in a well-mixed volume and that any interaction of the reactants has a fixed probability of resulting in a reaction, might use $X_i$ to describe the number of $A_i$ and have it evolve under a birth-death process with
    $$T(\vec{x})=\lambda\prod_i \frac{x_i!}{(x_i-r_i)!}$$</p>

    <blockquote>
        We call master equations describing these processes "chemical master equations".
    </blockquote>

    <p>Note that since the reaction notation uniquely specifies the transition
    rate $T(\vec{x})$, and the transition rate uniquely specifies the master
    equation, reaction notation can be used to precisely specify these kinds of
    birth-death processes.</p>
</section>

<section>
    <h2>Example: SIR model</h2>
    <p>In epidemiology, the stochastic SIR model (a <b>compartmental</b> model)
    can be described using the following pair of reactions:</p>
    \begin{align*}
    S + I &amp;\overset{\beta}{\longrightarrow} 2I\\
    I &amp;\overset{\mu}{\longrightarrow} R\\
    \end{align*}

    <p>The transition rates are $T_{inf}(n_S,n_I,n_R) = \beta n_S n_I$ with $\vec{v}_{inf}=(-1,1,0)$
    and $T_{rem}(n_S,n_I,n_R) = \mu n_R$ with $\vec{v}_{rem}=(0,-1,1)$.</p>
    <p>Master equation:
    $$\partial_t p(n_S,n_I,n_R)=\beta\left[(n_S+1)(n_I-1)p(n_S+1,n_I-1,n_R) - n_Sn_Ip(n_S,n_I,n_R)\right]\\
    + \mu\left[(n_I+1)p(n_S,n_I+1,n_R-1)-n_Ip(n_S,n_I,n_R)\right]$$</p>

    <p>Approximate mean dynamics:
    $$\partial_t \bar{n}_S = -\beta \bar{n}_S\bar{n}_I\\
    \partial_t \bar{n}_I = \beta \bar{n}_I\bar{n}_I - \mu\bar{n}_I\\
    \partial_t \bar{n}_R = \mu \bar{n}_I$$
    </p>
</section>

<section>
    <h2>Example: SIR model</h2>

    <div id="SIR_demo" style="height:500px">
    <script>
$(document).ready(function() {

    function runSIRDemo() {
        traj = SIR.getStochastic(50);
        trajDet = SIR.getDeterministic(50, 200);

        var traceS = {
            x: traj.t,
            y: traj.S,
            mode: "lines",
            name: "n_S",
            line: {
                color: "green",
                width: 2,
                shape: "hv"
            }
        };


        var traceSdet = {
            x: trajDet.t,
            y: trajDet.S,
            mode: "lines",
            name: "<n_S> (approx)",
            line: {
                color: "green",
                width: 2,
                dash: "dash"
            }
        };

        var traceI = {
            x: traj.t,
            y: traj.I,
            mode: "lines",
            name: "n_I",
            line: {
                color: "red",
                width: 2,
                shape: "hv"
            }
        };

        var traceIdet = {
            x: trajDet.t,
            y: trajDet.I,
            mode: "lines",
            name: "<n_I> (approx)",
            line: {
                color: "red",
                width: 2,
                dash: "dash"
            }
        };


        var traceR = {
            x: traj.t,
            y: traj.R,
            mode: "lines",
            name: "n_R",
            line: {
                color: "blue",
                width: 2,
                shape: "hv"
            }
        };

        var traceRdet = {
            x: trajDet.t,
            y: trajDet.R,
            mode: "lines",
            name: "<n_R> (approx)",
            line: {
                color: "blue",
                width: 2,
                dash: "dash"
            }
        };

        var traceR0 = {
            x: traj.t,
            y: SIR.getR0(traj),
            yaxis: "y2",
            mode: "lines",
            name: "R0",
            line: {
                color: "black",
                width: 2,
                shape: "hv"
            }
        };

        var traceR0det = {
            x: trajDet.t,
            y: SIR.getR0(trajDet),
            yaxis: "y2",
            mode: "lines",
            name: "<R0> (approx) ",
            line: {
                color: "black",
                width: 2,
                dash: "dash"
            }
        };

        var layout =  {
            width: 900,
            height: 500,
            margin: {
                t:0
            },
            xaxis: {
                title: "Time",
                range: [0, 50]
            },
            yaxis: {
                title: "Population size",
                range: [0, 101]
            },
            yaxis2: {
                title: "Effective reproductive number",
                overlaying: "y",
                side: "right",
                range: [0, 15]
            },
            legend: { x: 0.75, y: 0.75 }
        };

        var data = [traceS, traceSdet,
        traceI, traceIdet,
        traceR, traceRdet,
        traceR0, traceR0det];

        Plotly.newPlot("SIR_demo", data, layout, {displayModeBar: false});
    };

    $("#run_SIR").click(function(e) {
        e.preventDefault();

        runSIRDemo();
    });

    runSIRDemo();

});
    </script>
    </div>
    <a id="run_SIR" href="#">Simulate!</a> (Code <a href="SIR_demo.js">here</a>)
</section>

<section>
    <h2>What is population structure?</h2>

    <blockquote style="margin-top:2em">
        A populaion is structured if its members posess one or more traits that affect who they
        interact with. (Which reactions they are involved in.)
    </blockquote>

    <p style="margin-top:2em">Examples of characteristics that induce population structure:
    <ul class="spaced" style="width:80%">
        <li>Geographical location/position (phylogeography)</li>
        <li>Species membership (gene characteristic)</li>
        <li>Membership in epidemiological compartments
            <ul>
                <li>exposed vs infectious</li>
                <li>age category</li>
                <li>...</li>
            </ul></li>
    </ul>
    </p>
</section>

<section>
    <h2>Structured Wright-Fisher model</h2>

    <ul class="spaced" style="width:80%">
        <li>Introduced by <a class="cite" href="http://dx.doi.org/10.1007/BF00173909">Notohara, 1990</a> (and probably others)</li>
        <li>Applies the original Wright-Fisher model to structured populations.</li>
        <li>Assumes a single population is divided into subpopulations (demes) of size $N_i$ for $i\in {1\ldots D}$.</li>
        <li>Allows for weak migration between demes at rate $q_{ij}$.</li>
        <li>Sub-population sizes are unaffected by migration in the long term:
            imagine islands in an archipelago which have each equilibrated to their own carrying capacities.</li>
    </ul>
</section>

<section>
    <h2>Structured Wright-Fisher model</h2>

    <p>A $D=2$ structured WF generation proceeds as follows:</p>

    <img data-src="structWF.svg" style="width:70%">

    <ul>
        <li>$P(n'_{ba}|N_b)=\binom{n_b}{n'_{ba}}q_{ba}^{n'_{ba}}(1-q_{ba})^{n_b-n'_{ba}}$</li>
    </ul>
</section>

<section>
    <h2>The structured coalescent</h2>

    <div style="display:inline-block;width:49%;vertical-align:middle">
        <img data-src="structWF_genealogy.svg" style="width:100%">
    </div><div style="display:inline-block;width:49%;vertical-align:middle;text-align:left">
        <ul>
            <li>Backward in time genealogical process corresponding to taking
                the coalescent limit of the structured WF process.</li>
            <li>There are multiple routes to derivation! (It <b>can</b> be considered a
                structured WF process limit, but doesn't have to be.)</li>
        </ul>

        <dl style="margin-top:0.5em">
            <dt>Coalescence rate in deme $i$:</dt>
            <dd>$$T_{c}^{(i)}(\vec{k})=\binom{k_i}{2}\frac{1}{N_ig}$$</dd>
            <dt>Migration rate $i\rightarrow j$ (backward time):</dt>
            <dd>$$T_{m}^{(ij)}(\vec{k})=k_i m_{ij}$$</dd>
        </dl>

        <p>Note that $m_{ij} = g^{-1} q_{ji}\frac{N_j}{N_i}$.</p>
    </div>
</section>

<section>
    <h2>The effect of structure on tree shape</h2>

    <div style="width:100%;height:400px;position:relative">
    <img data-src="treeColours.png" style="width:100%;position:absolute;top:0;left:0">
    <img class="fragment" data-src="treeNoColours.png" style="width:100%;position:absolute;top:0;left:0">
    </div>

    <div class="fragment">
    <b>Signal for structure remains, even when tree is not coloured.</b>
    <blockquote class="fragment">
        Failure to account for structure can bias phylogenetic analyses.
    </blockquote>

    <div class="fragment"> (Trees simulated using <a href="StructuredCoalescentTree.xml">this MASTER script</a>.)</div>

</section>

<section>
    <h2>Expected time to coalesce</h2>

    <p>For a symmetric 2 deme model ($N_1=N_2=N$ and $m_{12}=m_{21}=m$) we can derive the expected time to coalescence of two lineages:</p>

    <ul>
        <li>Define $\tau_d$ and $\tau_s$ as coal. time when lineages are in different/same demes respectively.</li>
        <li>Lineages in distinct demes cannot coalesce: $$\langle\tau_d\rangle = \frac{1}{2m} + \tau_s$$</li>
        <li>Lineages in the same deme have probability $2m/(2m + 1/Ng)$ of migrating to distinct demes following the $1/(2m+1/Ng)$ waiting period:
            $$\langle\tau_s\rangle = \frac{1}{2m+1/Ng} + \frac{2m}{2m+1/Ng}\langle\tau_d\rangle$$<br></li>
        <li>Solving this recursion relation yields: $\langle\tau_s\rangle = 2Ng$ and $\langle\tau_d\langle = \frac{1}{2m} + 2Ng$</li>
    </ul>

    <blockquote>
        Expected time to coalesce from same deme is independent of $m$!  However, the variance is a (decreasing) function of $m$.
    </blockquote>
</section>

<section>
    <h2>Recap: Bayesian Phylogenetic Inference</h2>

    <p style="margin-top:10%">The usual phylogenetic posterior is:</p>
    <blockquote>
        $$P(T,\mu,\theta|A) = \frac{1}{P(A)}P(A|T,\mu)P(T|\theta)P(\mu)P(\theta)$$
    </blockquote>

    <p>where</p>
    <ul>
        <li>$A$ is the multiple sequence alignment,</li>
        <li>$T$ is the tree,</li>
        <li>$\mu$ are the substitution model parameters,</li>
        <li>$\theta$ are the tree prior parameters.</li>
    </ul>
</section>

<section>
    <h2>SC inference: Modified tree prior</h2>

    <p>The standard phylogenetic posterior is modified:</p>
    <blockquote>
        \begin{align*}
        P(T,C,\mu,\vec{N},m|A,L) = &amp;\frac{1}{P(A)}P(A|T,\mu)\\
        &amp;\times P(T,C|\vec{N},m,L)P(\mu)P(\theta)
        \end{align*}
    </blockquote>
    <ul>
        <li>$C$ are the tree colours (lineage locations),</li>
        <li>$L$ are the sample locations.</li>
    </ul>
    <p>The probability of a structured coalescent tree is:
    <div style="font-size:0.8em">
    \begin{align*}
    P(T,C|\vec{N},m,L)=&amp;\exp\left[-\int_0^{t_r}\left(\sum_i\binom{k_i}{2}\frac{1}{N_ig}+\sum_{ij}k_im_{ij}\right)\right]\\
    &amp;\times\left(\prod_{i}(N_ig)^{\nu_i}\right)\left(\prod_{ij}(m_{ij})^{\kappa_{ij}}\right)
    \end{align*}
    </div>
    ($\nu_i$ are the number of coalescences in $i$ and $\kappa_{ij}$ are
    the number of $i$ to $j$ migrations.)
    </p>

</section>

<section>
    <h2>Comparisson to mugration model</h2>

    <ul>
        <li>Alternative framework for phylogeographic inference.</li>
        <li>Introduced by <a class="cite" href="http://dx.doi.org/10.1371/journal.pcbi.1000520">Phillipe Lemey et al., 2009</a>.</li>
        <li>Treats location as "just another character" (migration $\leftrightarrow$ mutation)</li>
    </ul>

    <blockquote style="width:80%">
        \begin{align}
        P(T,\mu,\theta|A,L) =&amp; \frac{1}{P(A)P(L)} P(A|T,\mu)P(L|T,M)\\
        &amp;\times P(T|\theta)P(\mu)P(\theta)
        \end{align}
    </blockquote>
    <ul>
        <li>$L$ are the sampled locations, and</li>
        <li>$M$ is a matrix specifying the random walk.</li>
    </ul>

    <blockquote class="alert">
        Samples are assumed to be collected in a manner that is blind to their location.
    </blockquote>

    <ul>
        <li>Mugration models use sample location as data.</li>
        <li>Just as for genetic data, non-random sampling procedures will bias results.</li>
    </ul>

</section>


<section>
    <h2>Implementations of SC inference</h2>

    <ul class="spaced" style="margin-top:15%">
        <li>MIGRATE-n <a class="cite" href="http://www.genetics.org/cgi/content/abstract/185/1/313">(Beerli and Palczewski, 2010)</a></li>
        <li>Ewing sampler <a class="cite" href="http://www.genetics.org/content/168/4/2407.full-text.pdf%2Bhtml">(Ewing et al., 2004)</a></li>
        <li>MultiTypeTree (BEAST 2, <a class="cite" href="https://academic.oup.com/bioinformatics/article/doi/10.1093/bioinformatics/btu201/2748160/Efficient-Bayesian-inference-under-the-structured">Vaughan et al., 2014)</a></li>
        <li>BASTA (approximate, BEAST 2, <a class="cite" href="http://dx.doi.org/10.1371/journal.pgen.1005421">De Maio et al., 2015</a>)</li>
    </ul>
</section>

<section>
    <h2>Recommended reading</h2>

    <ul class="spaced" style="margin-top:15%">
        <li>Handbook of Stochastic Methods (Gardiner)</li>
        <li>Gene Genealogies, Variration and Evolution (Hein, Schierup, Wiuf)</li>
        <li>Inferring phylogenies (Felsenstein)</li>
    </ul>
</section>
